# Codebook

52 total points


```{r, include = FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
## DGE analysis overview

**.Rmd file = 01-DGE_setup_and_overview.Rmd**

### Setting up

#### Loading libraries; technically we don't need to load all of these libraries because they are in the file setup.R which is sourced in the first .Rmd of the module. However, it's good practice to load them all at the beginning of your script.
```{r, message = FALSE}

library(tidyverse)
library(RColorBrewer)
library(DESeq2)
library(pheatmap)
library(ggplot2)
library(ggrepel)

```

#### Loading data
```{r}

data <- read.delim("data/Mov10_full_counts.txt", row.names = 1)

meta <- read.delim("data/Mov10_full_meta.txt", row.names = 1)
```

Use class() to inspect our data and make sure we are working with data frames:
```{r}
### Check classes of the data we just brought in
class(meta)
class(data)
```

#### Viewing data

```{r}

rownames(meta)

names(data)

data[1:4,1:4]

```
You’ll notice that the colnames of the data and the rownames of metadata are the same except that they are in a different order. This is important later as we will be merging the metadata with the data based on these sample names.


### DGE analysis workflow

#### RNA-seq count distribution

To determine the appropriate statistical model, we need information about the distribution of counts. To get an idea about how RNA-seq counts are distributed, let’s plot the counts for a single sample, ‘Mov10_oe_1’:
```{r}
ggplot(data) +
geom_histogram(aes(x = Mov10_oe_1), stat = "bin", bins = 200) +
  xlab("Raw expression counts") +
  ylab("Number of genes")
```

If we zoom in close to zero, we can see a large number of genes with counts of zero:
```{r}
ggplot(data) +
geom_histogram(aes(x = Mov10_oe_1), stat = "bin", bins = 200) +
   xlim(-5, 500)  +
   xlab("Raw expression counts") +
   ylab("Number of genes")
```

These images illustrate some common features of RNA-seq count data, including a low counts associated with a large proportion of genes, and a long right tail due to the lack of any upper limit for expression.

#### Modeling count data

By plotting the *mean versus the variance* of our data we should be able to see that the variance > mean and therefore it does not fit the Poisson distribution and is better suited to the Negative Binomial (NB) model.

To calculate the mean and variance of our data, we will use the `apply` function. The `apply` function allows you to apply a function to the margins (rows or columns) of a matrix. The syntax for `apply` is as follows: `apply(X, MARGIN, FUN)`, where `X` is a matrix, `MARGIN` is the margin of the matrix to apply the function to (1 = rows, 2 = columns), and `FUN` is the function to apply. 

We will use `MARGIN = 1` to apply the mean and variance of the counts for each row (gene) across the 'Mov10 overexpression' replicates. We will then create a data frame with the mean and variance of the counts for each gene.
```{r}

mean_counts <- apply(data[, 3:5], 1, mean)

variance_counts <- apply(data[, 3:5], 1, var)

# for ggplot we need the data to be in a data.frame
df <- data.frame(mean_counts, variance_counts)
```

Run the following code to plot the *mean versus variance* for the 'Mov10 overexpression' replicates:
```{r}

ggplot(df) +
  geom_point(aes(x = mean_counts, y = variance_counts)) +
  geom_line(aes(x = mean_counts, y = mean_counts, color = "red")) +
  scale_y_log10() +
  scale_x_log10()
```

By plotting the mean versus the variance of our data, we can see that the variance is greater than the mean, indicating that the data does not fit the Poisson distribution and is better suited to the Negative Binomial (NB) model.

## Count normalization

**.Rmd file = 02-DGE_count_normalization.Rmd**

**Exercise** points +1

Determine the normalized counts for your gene of interest, PD1, given the raw counts and size factors below. 

NOTE: You will need to run the code below to generate the raw counts dataframe (PD1) and the size factor vector (size_factors), then use these objects to determine the normalized counts values:

```{r}

# Raw counts for PD1
PD1 <- c(21, 58, 17, 97, 83, 10)
names(PD1) <- paste0("Sample", 1:6)
PD1 <- data.frame(PD1)
PD1 <- t(PD1)

# Size factors for each sample
size_factors <- c(1.32, 0.70, 1.04, 1.27, 1.11, 0.85)

```

```{r}

# Your code here

```

Now that we know the theory of count normalization, we will normalize the counts for the Mov10 dataset using DESeq2. This requires a few steps:

1. Ensure the row names of the metadata dataframe are present and in the same order as the column names of the counts dataframe.
2. Create a `DESeqDataSet` object
3. Generate the normalized counts

### 1. Match the metadata and counts data

We should always make sure that we have sample names that match between the two files, and that the samples are in the right order. DESeq2 will output an error if this is not the case.

```{r}
### Check that sample names match in both files
all(colnames(data) %in% rownames(meta))
all(colnames(data) == rownames(meta))
```

The colnames of our data don't match the rownames of our metadata so we need to reorder them. We can use the `match` function:
```{r}


idx <- match(rownames(meta),colnames(data))
data <- data[,idx]

all(colnames(data) == rownames(meta))

```

**Exercise** points = +2

Suppose we had sample names matching in the counts matrix and metadata file, but they were out of order. Write the line(s) of code required to create a new matrix with columns ordered such that they were identical to the row names of the metadata.
```{r, eval = FALSE}

# write the code to reorder the columns of the 'counts' matrix to match the rownames of the metadata


# write the code to reorder the columns of the 'counts' matrix to match the rownames of the metadata


```


### 2. Create DESEq2 object

Bioconductor software packages often define and use a custom class within R for storing data (input data, intermediate data and also result data). These custom data structures are similar to `lists` in that they can contain multiple different data types/structures within them. But, unlike lists they have pre-specified `data slots`, which hold specific types/classes of data. The data stored in these pre-specified slots can be accessed by using specific package-defined functions.

Let's start by creating the `DESeqDataSet` object and then we can talk a bit more about what is stored inside it. To create the object we will need the **count matrix** and the **metadata** table as input. We will also need to specify a **design formula**. The design formula specifies the column(s) in the metadata table and how they should be used in the analysis. For our dataset we only have one column we are interested in, that is `~sampletype`. This column has three factor levels, which tells DESeq2 that for each gene we want to evaluate gene expression change with respect to these different levels.

```{r}
## Create DESeq2Dataset object
dds <- DESeqDataSetFromMatrix(countData = data, colData = meta, design = ~ sampletype)
```

You can use DESeq2-specific functions to access the different slots and retrieve information, if you wish. For example, suppose we wanted the original count matrix we would use `counts()`:

```{r}
head(counts(dds[,1:5]))
```

As we go through the workflow we will use the relevant functions to check what information gets stored inside our object. We can also run:

```{r}
slotNames(dds)
```

### 3. Generate the Mov10 normalized counts

The next step is to normalize the count data in order to be able to make fair gene comparisons between samples.

![Alt text](img/deseq_workflow_normalization_2018.png){ width=400 }

To perform the **median of ratios method** of normalization, DESeq2 has a single `estimateSizeFactors()` function that will generate size factors for us. We will use the function in the example below, but **in a typical RNA-seq analysis this step is automatically performed by the `DESeq()` function**, which we will see later. 

```{r}
dds <- estimateSizeFactors(dds)
```

By assigning the results back to the `dds` object we are filling in the slots of the `DESeqDataSet` object with the appropriate information. We can take a look at the normalization factor applied to each sample using:

```{r}
sizeFactors(dds)
```

Now, to retrieve the normalized counts matrix from `dds`, we use the `counts()` function and add the argument `normalized=TRUE`.

```{r}
normalized_counts <- counts(dds, normalized = TRUE)
```

We can save this normalized data matrix to file for later use:

```{r}
write.table(normalized_counts, file="data/normalized_counts.txt", sep="\t", quote=F, col.names=NA)
```

## Quality Control

**.Rmd file = 03-DGE_QC_analysis.Rmd**

**Exercise** points = +5

The figure below was generated from a time course experiment with sample groups 'Ctrl' and 'Sci' and the following timepoints: 0h, 2h, 8h, and 16h. 

![Alt text](img/PCA_example3.png){ width=600 }

**Determine the sources explaining the variation represented by PC1 and PC2.** 

- Ans: 


**Do the sample groups separate well?**

- Ans: 
  

**Do the replicates cluster together for each sample group?**

- Ans:

**Are there any outliers in the data?**

- Ans:
  

**Should we have any other concerns regarding the samples in the dataset?**

- Ans: 
  

**Example of Variance Stabilization**
Raw RNA-Seq count data has different levels of variability:

Without stabilization: Lowly expressed genes show high variability between samples (often due to noise), and highly expressed genes show less variability.
With stabilization (e.g., rlog or vst): Variance across genes is more evenly distributed, which makes the data easier to work with in analyses like clustering or PCA.

> The rlog transformation of the normalized counts is only necessary for these visualization methods during this quality assessment. 

```{r}
### Transform counts for data visualization
rld <- rlog(dds, blind=TRUE)
```
The `blind=TRUE` argument results in a transformation unbiased to sample condition information. When performing quality assessment, it is important to include this option. The [DESeq2 vignette](http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#blind-dispersion-estimation) has more details.

The `rlog` function returns a `DESeqTransform` object, another type of DESeq-specific object. The reason you don't just get a matrix of transformed values is because all of the parameters (i.e. size factors) that went into computing the rlog transform are stored in that object. We use this object to plot the PCA and heirarchical clustering figures for quality assessment.

### Principal components analysis (PCA)

DESeq2 has a built-in function for plotting PCA plots, that uses `ggplot2` under the hood. This is great because it saves us having to type out lines of code and having to fiddle with the different ggplot2 layers. In addition, it takes the `rlog` object as an input directly, hence saving us the trouble of extracting the relevant information from it.

The function `plotPCA()` requires two arguments as input: an `rlog` object and the `intgroup` (the column in our metadata that we are interested in). 

```{r, fig.width=6, fig.height=4, fig.align='center'}
### Plot PCA 
plotPCA(rld, intgroup="sampletype")
```


**What does this plot tell you about the similarity of samples? Does it fit the expectation from the experimental design?** By default the function uses the *top 500 most variable genes*. You can change this by adding the `ntop` argument and specifying how many genes you want to use to draw the plot.

### Hierarchical Clustering

Since there is no built-in function for heatmaps in DESeq2 we will be using the `pheatmap()` function from the `pheatmap` package. This function requires a matrix/dataframe of numeric values as input, and so the first thing we need to is retrieve that information from the `rld` object:

```{r}
### Extract the rlog matrix from the object
rld_mat <- assay(rld)    ## assay() is function from the "SummarizedExperiment" package that was loaded when you loaded DESeq2
```

Then we need to compute the pairwise correlation values for samples. We can do this using the `cor()` function:

```{r}
### Compute pairwise correlation values
rld_cor <- cor(rld_mat)    ## cor() is a base R function

head(rld_cor)   ## check the output of cor(), make note of the rownames and colnames
```

And now to plot the correlation values as a heatmap:

```{r, fig.width=6, fig.height=6, fig.align='center'}
### Plot heatmap
pheatmap(rld_cor)
```

Overall, we observe pretty high correlations across the board ( > 0.999) suggesting no outlying sample(s). Also, similar to the PCA plot you see the samples clustering together by sample group. Together, these plots suggest to us that the data are of good quality and we have the green light to proceed to differential expression analysis.

**Exercise** points = +3

The pheatmap function has a number of different arguments that we can alter from default values to enhance the aesthetics of the plot. Try adding the arguments `color`, `border_color`, `fontsize_row`, `fontsize_col`, `show_rownames` and `show_colnames` to your pheatmap. How does your plot change. Take a look through the help pages (?pheatmap) and identify what each of the added arguments is contributing to the plot.


```{r, fig.width=8, fig.height=8, echo=FALSE}
# Display all available color palettes
display.brewer.all()
```

```{r}
# The color palette "Blues" is a good choice for this heatmap
heat.colors <- brewer.pal(9, "Blues")

# Your code here
pheatmap(rld_cor, border_color = "grey", fontsize_row = 12, fontsize_col = 12, show_rownames = TRUE, show_colnames = TRUE, color = heat.colors)
```

## DGE analysis workflow

**.Rmd file = 04_DGE_DESeq2_analysis.Rmd**

### Set the Design Formula

The main purpose of the **design formula** in DESeq2 is to specify the factors that are influencing gene expression so that their effects can be **accounted for** or **controlled** during the analysis. This allows DESeq2 to isolate the effect of the variable you're primarily interested in while adjusting for other known sources of variation. The design formula should have all of the factors in your metadata that account for major sources of variation in your data. The last factor entered in the formula should be the condition of interest.

For example:
```{r}
design(dds)
```
This shows that **sampletype** is the condition of interest and as such is the only covariate. DESeq2 will test for differential expression between the two sample types, adjusting for any other variation.

**Exercise** points = +3

1. Suppose you wanted to study the expression differences between the two age groups in the metadata shown above, and major sources of variation were `sex` and `treatment`, how would the design formula be written?

```{r, eval = FALSE}

# the default behavior of `results` is to return the comparison of the last group in the design formula, so your condition of interest should come last:


```

2. Based on our Mov10 metadata dataframe, which factors could we include in our design formula?

- Ans:

3. What would you do if you wanted to include a factor in your design formula that is not in your metadata?

- Ans:

### MOV10 Differential Expression Analysis


**Running Differential Expression in Two Lines of Code**

To obtain differential expression results from our raw count data, we only need to run two lines of code!

First, we create a DESeqDataSet, as we did in the ['Count normalization'](https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html#2-create-deseq2-object) lesson, specifying the location of our raw counts and metadata, and applying our design formula:

```{r}
## Create DESeq object
dds <- DESeqDataSetFromMatrix(countData = data, colData = meta, design = ~ sampletype)
```

Next, we run the actual differential expression analysis with a single call to the `DESeq()` function. This function handles everything—from **normalization** to **linear** modeling—all in one step. During execution, `DESeq2` will print messages detailing the steps being performed: estimating size factors, estimating dispersions, gene-wise dispersion estimates, modeling the mean-dispersion relationship, and statistical testing for differential expression.

```{r}
## Run analysis
dds <- DESeq(dds)
```

By re-assigning the result to back to the same variable name (`dds`), we update our `DESeqDataSet` object, which will now contain the results of each step in the analysis, effectively filling in the `slots` of our `DESeqDataSet` object.

### DESeq2 differential gene expression analysis workflow

Everything from normalization to linear modeling was carried out by the use of a single function!

With the 2 lines of code above, we just completed the workflow for the differential gene expression analysis with DESeq2. The steps in the analysis are output below:

![Alt text](img/deseq2_workflow_separate.png){ width=200 }

#### Step 1: Estimate size factors

MOV10 DE analysis: examining the size factors
```{r}
## Check the size factors
sizeFactors(dds)
```

Take a look at the total number of reads for each sample:
```{r}
## Total number of raw counts per sample
colSums(counts(dds, normalized = FALSE))
```

**How do the numbers correlate with the size factor?**

- Ans: 

### MOV10 Differential Expression Analysis: Exploring Dispersion Estimates

Now, let's explore the dispersion estimates for the MOV10 dataset:

```{r}
## Plot dispersion estimates
plotDispEsts(dds)
```

**Since we have a small sample size, for many genes we see quite a bit of shrinkage. 

**Exercise** points = +1

**Do you think our data are a good fit for the model?**

- Ans: 

## Negative Binomial model fitting

**.Rmd file = 05_DGE_DESeq2_analysis2.Rmd**


3. **Estimate the Beta Coefficients:**
"DESeq2 estimates the **beta coefficients** by fitting a **negative binomial GLM** to the data using **Maximum Likelihood Estimation (MLE)**. DESeq2 substitutes the mean (\( \mu_{ij} \)) with \( 2^{\beta_i X_j} \) in the negative binomial model, using MLE to predict expected counts based on experimental covariates and estimate gene expression changes while controlling for design factors.

The model is fit using the **`DESeq()`** function in R, which carries out the entire estimation process:

```{r}
dds <- DESeq(dds)
```

4. **Extract the Beta Coefficients:**
Once the model is fitted, the estimated beta coefficients (log2 fold changes) for each gene can be extracted using the **`coef()`** function:

```{r}
beta_coefficients <- coef(dds)
```

This gives a matrix of estimated beta coefficients, where each row corresponds to a gene and each column corresponds to a covariate in the design matrix.

### Example:
For example, if the design matrix includes an intercept (control) and two conditions (MOV10 knockdown and MOV10 overexpression), the beta coefficients \( \beta_i \) for gene \( i \) would represent:

- \( \beta_{i1} \): the log2 fold change for the control group (intercept)

- \( \beta_{i2} \): the log2 fold change for the MOV10 knockdown group

- \( \beta_{i3} \): the log2 fold change for the MOV10 overexpression group.

### Calculation:
To calculate the fitted (expected) log2 counts for **gene \( i \)** in **sample \( j \)**, you can take the **dot product** of the row of the design matrix \( X_j \) for that sample and the beta coefficients \( \beta_i \) for that gene:

$$
\log_2(\mu_{ij}) = \beta_i X_j = \beta_{i1} X_{j1} + \beta_{i2} X_{j2}  + \dots + \beta_{iP} X_{jP}
$$

Where:

- \( P \) is the number of covariates in the design matrix

The expected mean \( \mu_{ij} \) on the original count scale is obtained by exponentiating the log2-scale fitted values:

$$
\mu_{ij} = 2^{\beta_i X_j}
$$

**Exercise** points = +1

In the DESeq2 workflow, what is the purpose of the design matrix in the context of fitting the Negative Binomial model for each gene?

- Ans:


### Matrix Multiplication Form:
If you want to calculate the fitted values for all genes and samples at once, you can express the model as a **matrix multiplication**:

$$
\log_2(\mu) = \beta X^T
$$

This computes the expected log2 counts for all genes across all samples.

In R, you can calculate the fitted log2 counts for all genes and samples using:

```{r}

log2_fitted <- beta_coefficients %*% t(X)
``` 

This step multiplies the beta coefficients matrix by the transpose of the design matrix to compute the fitted log2 counts.

STOP Tues Oct 22

## MOV10 Differential Expression Analysis:

Control versus Overexpression

**Building the Results Table**

To build the results table, we use the `results()` function. You can specify the `contrast` to be tested using the contrast argument. In this example, we’ll save the unshrunken and shrunken results of **Control vs. Mov10 overexpression** to different variables. We’ll also set the `alpha` to 0.05, which is more stringent than the default value of 0.1.
```{r}
# define contrasts
contrast_oe <- c("sampletype", 
                 "MOV10_overexpression",
                 "control")

# extract results table
res_tableOE_unshrunken <- results(dds, 
                          contrast = contrast_oe, 
                          alpha = 0.05)

resultsNames(dds)

# shrink log2 fold changes
res_tableOE <- lfcShrink(dds = dds, 
  coef =  "sampletype_MOV10_overexpression_vs_control", 
  res = res_tableOE_unshrunken)

# save the results for future use
saveRDS(res_tableOE, file = "data/res_tableOE.RDS")

```

**MA Plot**

A **MA plot** visualizes the relationship between the **mean expression** (on the x-axis) and the **log2 fold change** (on the y-axis) for all genes in a dataset. Genes that are significantly differentially expressed are highlighted, while most other genes cluster around zero log2 fold change. 

When **shrinkage** is applied, as in DESeq2, large log2 fold change estimates for genes with low counts or high variability are pulled toward more moderate values, reducing the noise in the plot. This helps focus attention on the more reliable fold changes and makes the plot cleaner, especially for low-expression genes that tend to have more variable LFC estimates without shrinkage.

This plot is also a great way to illustrate the effect of LFC shrinkage. DESeq2 provides a simple function to generate an MA plot.

**Shrunken & Unshrunken Results:**
```{r}

par(mfrow = c(1,2))

plotMA(res_tableOE_unshrunken, ylim=c(-2,2))

abline(v = 10,col="red",lwd = 2)

# Shrunken results:
plotMA(res_tableOE, ylim=c(-2,2))

abline(v = 10,col="red",lwd = 2)

```

**MOV10 DE Analysis: Exploring the Results**

The results table in `DESeq2` looks similar to a data.frame and can be treated like one for accessing or subsetting data. However, it is stored as a `DESeqResults` object, which is important to keep in mind when working with visualization tools.
```{r}
class(res_tableOE)
```

Let's go through some of the columns in the results table to get a better idea of what we are looking at. To extract information regarding the meaning of each column we can use `mcols()`:
```{r}
mcols(res_tableOE, use.names = T)
```

Now let's take a look at what information is stored in the results:
```{r}
head(res_tableOE)
names(res_tableOE)
```

### MOV10 DE Analysis: **Control vs. Knockdown**

After examining the overexpression results, let's move on to the comparison between **Control vs. Knockdown**. We’ll use contrasts in the `results()` function to extract the results table and store it in the `res_tableKD` variable. You can also use `coef` to specify the contrast directly in the `lfcShrink()` function.
```{r}

# define contrast
contrast_kd <-  c("sampletype", 
                  "MOV10_knockdown",
                  "control")

# extract results table
res_tableKD_unshrunken <- results(dds, 
                                  contrast = contrast_kd, 
                                  alpha = 0.05)

resultsNames(dds)


# use the `coef` argument to specify the contrast directly
res_tableKD <- lfcShrink(dds, coef = 2, 
                         res = res_tableKD_unshrunken)

## Save results for future use
saveRDS(res_tableKD, file = "data/res_tableKD.RDS")

```


## Summarizing Results

To summarize the results, DESeq2 offers the `summary()` function, which conveniently reports the number of genes that are significantly differentially expressed at a specified threshold (`default FDR < 0.05`). Note that, even though the output refers to p-values, it actually summarizes the results using **adjusted p-values (padj/FDR)**.

Let’s start by summarizing the results for the OE vs. control comparison:
```{r}
## Summarize results
summary(res_tableOE)
```

In addition to reporting the number of up- and down-regulated genes at the default significance threshold, this function also provides information on:

- **Number of genes tested** (genes with non-zero total read count)
- **Number of genes excluded** from multiple test correction due to low mean counts

### Extracting Significant Differentially Expressed Genes

In some cases, using only the FDR threshold doesn’t sufficiently reduce the number of significant genes, making it difficult to extract biologically meaningful results. To increase stringency, we can apply an additional fold change threshold.

Although the summary() function doesn't include an argument for fold change thresholds, we can define our own criteria.

Let’s start by setting the thresholds for both adjusted p-value (FDR < 0.05) and log2 fold change (|log2FC| > 0.58, corresponding to a 1.5-fold change):

```{r}
### Set thresholds
padj.cutoff <- 0.05
lfc.cutoff <- 0.58
```

Next, we’ll convert the results table to a tibble for easier subsetting:

```{r}
res_tableOE_tb <- res_tableOE %>%
  data.frame() %>%
  rownames_to_column(var = "gene") %>% 
  as_tibble()
```

Now, we can filter the table to retain only the genes that meet the significance and fold change criteria:

```{r}

sigOE <- res_tableOE_tb %>%
  dplyr::filter(padj < padj.cutoff & 
                abs(log2FoldChange) > lfc.cutoff)

# Save the results for future use
saveRDS(sigOE,"data/sigOE.RDS")

```

**Exercise** points = +2 

How many genes are differentially expressed in the **Overexpression vs. Control** comparison based on the criteria we just defined? Does this reduce the number of significant genes compared to using only the FDR threshold?
```{r}

# Your code here

```

  
**Does this reduce our results?** 
```{r}

# Your code here

```


### MOV10 Knockdown Analysis: Control vs. Knockdown

Next, let’s perform the same analysis for the **Control vs. Mov10 knockdown** comparison. We’ll use the same thresholds for adjusted p-value (FDR < 0.05) and log2 fold change (|log2FC| > 0.58).
```{r}

res_tableKD_tb <- res_tableKD %>%
  data.frame() %>%
  rownames_to_column(var = "gene") %>%
  as_tibble()

sigKD <- res_tableKD_tb %>%
         dplyr::filter(padj < padj.cutoff & 
         abs(log2FoldChange) > lfc.cutoff)

# We'll save this object for use in the homework
saveRDS(sigKD,"data/sigKD.RDS")
```

**How many genes are differentially expressed in the Knockdown compared to Control?**
```{r}
# Your code here
```

## Visualizing RNA-seq results

**.Rmd file = 06_DGE_visualizing_results.Rmd**

Let’s start by loading a few libraries (if not already loaded):
```{r}

# load libraries
library(tidyverse)
library(ggplot2)
library(ggrepel)
library(RColorBrewer)
library(DESeq2)
library(pheatmap)
library(dplyr)

```

When we are working with large amounts of data it can be useful to display that information graphically to gain more insight.

Let’s create tibble objects from the meta and normalized_counts data frames before we start plotting. This will enable us to use the tidyverse functionality more easily.
```{r}

# Read in the metadata
meta <- read.table("data/Mov10_full_meta.txt", 
                   header=T, row.names=1)

# Create a tibble for meta data
mov10_meta <- meta %>% 
  rownames_to_column(var="samplename") %>% 
  as_tibble()
        
# read in the normalized counts
normalized_counts <- read.delim("data/normalized_counts.txt", row.names=1)

all(mov10_meta$samplename == colnames(normalized_counts))

# Create a tibble for normalized_counts
normalized_counts <- normalized_counts %>% 
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble()
```

One way to visualize results would be to simply plot the expression data for a handful of genes. We could do that by picking out specific genes of interest or selecting a range of genes.

#### **Using DESeq2 `plotCounts()` to plot expression of a single gene**

To pick out a specific gene of interest to plot, for example Mov10, we can use the `plotCounts()` from DESeq2. `plotCounts()` is a function that allows us to plot the normalized counts for a single gene across all samples.

```{r}

plotCounts(dds, 
           gene="MOV10", 
           intgroup="sampletype") 

# We can also color the points by sample type
sampletype = as.factor(mov10_meta$sampletype)

library(RColorBrewer)

display.brewer.all()

col = brewer.pal(8,"Dark2")
palette(col)

plotCounts(dds, gene="MOV10", 
           intgroup="sampletype",
           col = as.numeric(sampletype),
           pch = 19) 

```

**This function only allows for plotting the counts of a single gene at a time.** 

#### **Using ggplot2 to plot expression of a single gene**

We can also use ggplot2 to plot the `MOV10` counts. We can save the output of `plotCounts()` to a variable specifying the `returnData=TRUE` argument. This will save the normalized counts for the gene `MOV10` to a data frame object. We can then use ggplot2 to plot the normalized counts for `MOV10` across all samples.

```{r}
# Save `plotCounts()` to a data frame object
d <- plotCounts(dds, gene = "MOV10", intgroup = "sampletype", 
                returnData=TRUE)

# Plot using ggplot2
ggplot(d, aes(x = sampletype, y = count, color = sampletype)) + 
  geom_point(position = position_jitter(w = 0.1,h = 0)) +
  geom_text_repel(aes(label = rownames(d))) + 
  theme_bw() +
  ggtitle("MOV10") +
  theme(plot.title = element_text(hjust = 0.5))
```

> Note that in the plot below (code above), we are using `geom_text_repel()` from the `ggrepel` package to label our individual points on the plot.



#### Using `ggplot2` to plot multiple genes (e.g. top 20)

Often it is helpful to check the expression of multiple genes of interest at the same time. This often first requires some data wrangling.

We are going to plot the normalized count values for the **top 20 differentially expressed genes (by padj values)**. 

To do this, we first need to determine the gene names of our top 20 genes by ordering our results and extracting the top 20 genes (by padj values):

```{r}

res_tableOE <- readRDS("data/res_tableOE.RDS")

res_tableOE_tb <- res_tableOE %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble()

## Order results by padj values
top20_sigOE_genes <- res_tableOE_tb %>% 
        arrange(padj) %>% 	
#Arrange rows by padj values
        pull(gene) %>% 		
#Extract character vector of ordered genes
        head(n=20) 		
#Extract the first 20 genes
```

Then, we can extract the normalized count values for these top 20 genes:

```{r}
## Normalized counts for top 20 significant genes
top20_sigOE_norm <- normalized_counts %>%
        filter(gene %in% top20_sigOE_genes)
```

Now that we have the normalized counts for each of the top 20 genes for all 8 samples, to plot using `ggplot()`, we need to `pivot_longer` top20_sigOE_norm from a wide format to a long format so the counts for all samples will be in a single column to allow us to give ggplot the one column with the values we want it to plot.

The `pivot_longer()` function in the **tidyr** package will perform this operation and will output the normalized counts for all genes for *Mov10_oe_1* listed in the first 20 rows, followed by the normalized counts for *Mov10_oe_2* in the next 20 rows, so on and so forth.

![Alt text](img/melt_wide_to_long_format.png){ width=800 }

```{r}
# Pivot the data frame
pivoted_top20_sigOE <- top20_sigOE_norm %>%
  pivot_longer(colnames(top20_sigOE_norm)[2:9], 
               names_to = "samplename", 
               values_to = "normalized_counts")

## Check the column header in the "pivoted" data frame
head(pivoted_top20_sigOE)
```

Now, if we want our counts colored by sample group, then we need to combine the metadata information with the melted normalized counts data into the same data frame for input to `ggplot()`:

`inner_join(x,y)` will merge 2 data frames by the colname in x that matches a column name in y in this case `samplename` column.
```{r}

pivoted_top20_sigOE <- inner_join(
  mov10_meta, 
  pivoted_top20_sigOE)
```


Now that we have a data frame in a format that can be utilised by ggplot easily, let's plot! 


```{r}
## plot using ggplot2
ggplot(pivoted_top20_sigOE) +
        geom_point(aes(x = gene, y = normalized_counts, 
          color = sampletype)) +
        scale_y_log10() +
        xlab("Genes") +
        ylab("log10 Normalized Counts") +
        ggtitle("Top 20 Significant DE Genes") +
        theme_bw() +
      	theme(axis.text.x = element_text(angle = 45, 
      	                                 hjust = 1)) +
      	theme(plot.title = element_text(hjust = 0.5))
```


### Heatmap

We will plot a heatmap of the `sigOE` genes to visualize their expression across all samples using `pheatmap()`.
```{r}

sigOE = readRDS("data/sigOE.RDS")

norm_OEsig <- normalized_counts[,c(1,2:4,7:9)] %>% 
              filter(gene %in% sigOE$gene) %>% 
      	      data.frame() %>%
      	      column_to_rownames(var = "gene") 

```

Now let’s draw the heatmap using pheatmap. We can also add annotations to the heatmap to indicate the sample type. The `annotation_col` argument in `pheatmap()`requires a data frame with the same row names as the column names of the matrix used to generate the heatmap. 
```{r}
### Annotate our heatmap (optional)

annotation <- mov10_meta %>% 
  filter(samplename %in% colnames(norm_OEsig)) %>%
	dplyr::select(samplename, sampletype) %>% 
	data.frame(row.names = "samplename")

annotation$sampletype = factor(annotation$sampletype)

### Set a color palette
heat_colors <- brewer.pal(6, "YlOrRd")

### Set annotation colors
ann_colors = list(
  sampletype = c(control="navy",
  MOV10_overexpression= "green"))

### Run pheatmap
pheatmap(norm_OEsig, 
         annotation_colors = ann_colors, 
         color = heat_colors, 
         cluster_rows = T, 
         show_rownames = F,
         annotation_col = annotation, 
         border_color = NA, 
         fontsize = 12, 
         scale = "row", 
         fontsize_col = 12)
```


> *NOTE:* There are several additional arguments we have included in the function for aesthetics. One important one is `scale="row"`, in which Z-scores are plotted, rather than the actual normalized count value. 

### Volcano plots

A commonly used plot in RNA-seq analysis is the **volcano plot**. in which you have the log transformed adjusted p-values plotted on the y-axis and log2 fold change values on the x-axis. 

To generate a volcano plot, we first need to have a column in our results data indicating whether or not the gene is considered differentially expressed based on p-adjusted values.
```{r}
## create a column for thresholding in the results table
res_tableOE_tb <- res_tableOE_tb %>% 
                  mutate(threshold_OE = padj < 0.05 & 
                  abs(log2FoldChange) >= 0.58)
```

Now we can start plotting. The `geom_point` object is most applicable, as this is essentially a scatter plot:
```{r}

ggplot(res_tableOE_tb) +
        geom_point(aes(x = log2FoldChange, 
        y = -log10(padj), colour = threshold_OE)) +
        ggtitle("Mov10 overexpression") +
        xlab("log2 fold change") + 
        ylab("-log10 adjusted p-value") +
        theme_bw() +      
        theme(legend.position = "none",
        plot.title = element_text(size = rel(1.5), 
                                  hjust = 0.5),
        axis.title = element_text(size = rel(1.25)))  
```

What if we also wanted to know where the top 10 genes (lowest padj) in our DE list are located on this plot? We could label those dots with the gene name on the Volcano plot using `geom_text_repel()`.

First, we need to order the res_tableOE tibble by `padj`, and add an additional column to it, to include on those gene names we want to use to label the plot.
 
```{r}
## Create a column to indicate which genes to label
res_tableOE_tb <- res_tableOE_tb %>% 
  arrange(padj) %>% 
  mutate(genelabels = "")

res_tableOE_tb$genelabels[1:10] <- res_tableOE_tb$gene[1:10]

head(res_tableOE_tb)
```

Next, we plot it as before with an additiona layer for `geom_text_repel()` wherein we can specify the column of gene labels we just created. 

```{r}
ggplot(res_tableOE_tb, aes(x = log2FoldChange, 
        y = -log10(padj))) +
        geom_point(aes(colour = threshold_OE)) +
        geom_text_repel(aes(label = genelabels)) +
        ggtitle("Mov10 overexpression") +
        xlab("log2 fold change") + 
        ylab("-log10 adjusted p-value") +
        theme_bw() +
        theme(legend.position = "none",
        plot.title = element_text(size = rel(1.5), 
                                  hjust = 0.5),
        axis.title = element_text(size = rel(1.25))) 
```

## Summary of DGE workflow

**.Rmd file = 07-DGE_summarizing_workflow.Rmd**

```{r, message = FALSE}
## Setup
library(DESeq2)
```
We have detailed the various steps in a differential expression analysis workflow, providing theory with example code. To provide a more succinct reference for the code needed to run a DGE analysis, we have summarized the steps in an analysis below:

### 1. Import data into dds object:
```{r, eval = FALSE}

# check that the rownames of the count matrix 
# match the colnames of the metadata
all(colnames(raw_counts) == rownames(metadata))

# Create DESeq2Dataset object
dds <- DESeqDataSetFromMatrix(countData = raw_counts, colData = metadata, 
                              design = ~ condition)

```

### 2. Exploratory data analysis (PCA & heirarchical clustering) - identifying outliers and sources of variation in the data:
```{r, eval = FALSE}

# Transform counts for data visualization
rld <- rlog(dds, blind = TRUE)

# Plot PCA
plotPCA(rld, intgroup = "sampletype")

# Extract the rlog matrix from the object
rld_mat <- assay(rld)

# Compute pairwise correlation values
rld_cor <- cor(rld_mat)

# Plot heatmap
pheatmap(rld_cor)
```

### 3. Run DESeq2:
```{r, eval = FALSE}

# **Optional step** - Re-create DESeq2 dataset if the design formula has changed after QC analysis in include other sources of variation

dds <- DESeqDataSetFromMatrix(countData = raw_counts, 
                              colData = metadata, design = ~ condition)

# Run DESeq2 differential expression analysis
dds <- DESeq(dds)

# Output normalized counts to save as a file to access outside RStudio
normalized_counts <- counts(dds, normalized = TRUE)
    
write.table(normalized_counts, file = "data/normalized_counts.txt", 
            sep = "\t", quote = F, col.names = NA)

```

### 4. Check the fit of the dispersion estimates:
```{r, eval = FALSE}

# Plot dispersion estimates
plotDispEsts(dds)
```

### 5. Create contrasts to perform Wald testing on the shrunken log2 foldchanges between specific conditions:
```{r, eval = FALSE}

# Output results of Wald test for contrast
contrast <- c("condition", 
              "level_to_compare", 
              "base_level")

res <- results(dds, contrast = contrast)

coef = resultsNames(dds)

res_table <- lfcShrink(dds, 
                       coef = coef[2], 
                       res = res, 
                       type = "apeglm")
```

### 6. Output significant results:
```{r, eval = FALSE}
### Set thresholds
padj.cutoff <- 0.05
lfc.cutoff <- 0.58 ## change in expression of 1.5

# Turn the results object into a data frame
res_df <- res %>%
  data.frame() %>%
  rownames_to_column(var = "gene") 

# Subset the significant results
sig_res <- dplyr::filter(res_df, 
                  padj < padj.cutoff &  
                  abs(log2FoldChange) > lfc.cutoff)
```

### 7. Visualize results: volcano plots, heatmaps, normalized counts plots of top genes, etc.

### 8. Make sure to output the versions of all tools used in the DE analysis:
```{r}
sessionInfo()
```

**Exercise/Homework: modify the `lessons/07-DGE_summarizing_workflow.Rmd` file to analyze the MOV dataset, starting with Mov10_full_counts.txt in your data folder. Compare the "MOV10_knockdown" to the "control". Include a heatmap and a volcano plot**. Make sure you rename the file first to `Lastname_07-DGE_summarizing_workflow.Rmd` and save it in your main directory, not in the lessons folder. points = +10

## Functional analysis of RNA-seq data

**.Rmd file = 08-GO_enrichment_analysis.Rmd**

### Over-representation Analysis (ORA) using `clusterProfiler`
```{r, message=FALSE}


# Uncomment the following if you haven't yet installed the required packages
# BiocManager::install(c("ggraph", "igraph", "visNetwork", 
# "GO.db", "GOSemSim"), force = TRUE)

# IMPORTANT!!! Uncomment the following to update the GOenrichment package
# remove.package("GOenrichment")
# devtools::install_github("gurinina/GOenrichment", force = TRUE)

## Load required libraries
library(org.Hs.eg.db)#
library(clusterProfiler)#
library(tidyverse)
library(enrichplot)#
library(fgsea) #
library(igraph)
library(ggraph)
library(visNetwork)
library(GO.db)
library(GOSemSim)
library(GOenrichment)
```

#### Running clusterProfiler: we first need to load the results of the differential expression analysis.
```{r}

res_tableOE = readRDS("data/res_tableOE.RDS")

res_tableOE_tb <- res_tableOE %>% 
  data.frame() %>% rownames_to_column(var = "gene") %>% 
  dplyr::filter(!is.na(log2FoldChange))  %>% as_tibble()

```

To perform the over-representation analysis, we need a list of background genes and a list of significant genes:
```{r}

## background set of genes
allOE_genes <- res_tableOE_tb$gene


sigOE = dplyr::filter(res_tableOE_tb, padj < 0.05)

## significant genes
sigOE_genes = sigOE$gene

```


The`enrichGO()` function performs the ORA for the significant genes of interest (`sigOE_genes`) compared to the background gene list (`allOE_genes`) and returns the enriched GO terms and their associated p-values. 
```{r}

## Run GO enrichment analysis
ego <- enrichGO(gene = sigOE_genes,
 universe = allOE_genes,
 keyType = "SYMBOL",
 OrgDb = org.Hs.eg.db,
 minGSSize = 20,
 maxGSSize = 300,
 ont = "BP",
 pAdjustMethod = "BH",
 qvalueCutoff = 0.05,
 readable = TRUE)

```


We can extract the gene lists associated with each enriched GO term from the `ego` object to create edges between terms based on shared genes.
```{r}
# Extract the result as a data frame
enrich_df <- as.data.frame(ego)

names(enrich_df)
head(enrich_df)

# GO term IDs 
term_ids <- enrich_df$ID  

# Split gene lists by "/" and assign term IDs as names
gene_lists <- strsplit(enrich_df$geneID, "/") 

# Assign term IDs as names
names(gene_lists) <- term_ids  

```

Using the gene lists associated with each GO term, we can create edges between terms based on the shared genes.
```{r}

# Load GO data for semantic similarity calculation
go_data <- GOSemSim::godata(OrgDb = "org.Hs.eg.db", 
                            ont = "BP", 
                            keytype = "SYMBOL",
                            computeIC = TRUE)
            

# Compute semantic similarity
similarity_matrix <- GOSemSim::mgoSim(term_ids, term_ids, 
                     semData = go_data, 
                     measure = "Wang", combine = NULL)

# Convert similarity matrix to edges
edges <- data.frame(from = rep(term_ids, 
                    each = length(term_ids)),
                    to = rep(term_ids, 
                    times = length(term_ids)),
                    similarity = as.vector(similarity_matrix))

# Filter out edges with zero similarity or self-loops
edges <- subset(edges, similarity > 0 & from != to)

edges <- edges %>% 
  dplyr::filter(similarity >= 0.5)


# Convert term IDs to names
edges$to <- Term(edges$to)
edges$from <- Term(edges$from)

head(edges)

```

Now we can create a network plot to visualize the relationships between the enriched GO terms based on their semantic similarity.
```{r}
# Create an igraph object from the edges data frame
g <- graph_from_data_frame(edges, directed = FALSE)

# Visualize the graph using ggraph
ggraph(g, layout = 'fr') +  # Fruchterman-Reingold layout
  geom_edge_link(aes(edge_alpha = 0.5)) +  # Edges with some transparency
  geom_node_point(size = 5, color = "blue") +  # Nodes sized uniformly
  geom_node_text(aes(label = name), repel = TRUE, size = 3) +  # Node labels
  theme_void() +  # Clean theme without axes
  labs(title = "GO Terms Semantic Similarity Network")  # Title for the plot

```

Edges based can also be defined by Jaccard similarity; defined as the size of the intersection each pair of GO terms divided by the size of their union 
```{r}

# Define a function to compute Jaccard similarity
jaccard_index <- function(genes1, genes2) {
length(intersect(genes1, genes2)) / length(union(genes1, genes2))
}

# Create an empty data frame to store edges
edges <- data.frame(from = character(), to = character(), 
                    similarity = numeric())
  # Compute Jaccard similarity for each pair of terms
  for (i in 1:(length(term_ids) - 1)) {
  for (j in (i + 1):length(term_ids)) {
  similarity <- jaccard_index(gene_lists[[i]], gene_lists[[j]])
  if (similarity > 0) {  # Only keep edges with some overlap
  edges <- rbind(edges, data.frame(from = term_ids[i], 
                                   to = term_ids[j], 
                                   similarity = similarity))
  }}}

edges <- edges %>% 
  dplyr::filter(similarity >= 0.5)

# Convert term IDs to names
edges$to <- Term(edges$to)
edges$from <- Term(edges$from)

```

Now we can create a network plot to visualize the relationships between the enriched GO terms based on the Jaccard similarity of their gene lists.
```{r}
# Create a graph from the edges
g <- graph_from_data_frame(edges, directed = FALSE)

# Visualize the graph using ggraph
ggraph(g, layout = 'fr') +  # Fruchterman-Reingold layout
  geom_edge_link(aes(edge_alpha = 0.5)) +  # Edges with some transparency
  geom_node_point(size = 5, 
                  color = "blue") + 
  geom_node_text(aes(label = name), 
                 repel = TRUE, 
                 size = 3) +  # Node labels
  theme_void() +  # Clean theme without axes
  labs(title = "GO Terms Jaccard Overlap Network") 

```

The resulting network plots shows the relationships between the enriched GO terms based on either Semantic or Jaccard similarity of their gene lists. 

### Visualizing Enrichment Results

**Network Plot**:
The `emapplot()` function generates a network plot where each node represents an enriched gene set, and edges between nodes indicate the similarity or overlap between those gene sets. 
```{r}

pwt <- pairwise_termsim(
  ego,
  method = "JC",
  semData = NULL
)

# category labels
emapplot(pwt, showCategory = 30, 
         node_label = "category", 
         cex_label_group = 3, 
         layout.params = list("mds")) +
         theme(legend.position = "none")

# group labels 
emapplot(pwt, showCategory = 150, 
         node_label = "group") +
         theme(legend.position = "none")

```
To save any of these figures if they become too cluttered, click on the Export button in the RStudio Plots tab and Save as PDF. In the pop-up window, change the PDF size to 24 x 32 to give a figure of appropriate size for the text labels.

**cnetplot**: Visualizes connections between genes and top GO terms, highlighting shared pathways.This plot is particularly useful for hypothesis generation in identifying genes that may be important to several of the most affected processes.
```{r}

## Extract the foldchanges
OE_foldchanges <- sigOE$log2FoldChange

names(OE_foldchanges) <- sigOE$gene

## cnetplot for the top 5 categories
cnetplot(ego,
  categorySize="padj",
  node_label = "all",
  showCategory = 5,
  foldChange = OE_foldchanges,
  vertex.label.font = 6)

```

### GSEA Using clusterProfiler and fgsea

Prepare fold changes, sort by expression, and run GSEA with `gseGO()` for clusterProfiler or `fgseaSimple()` for `fgsea`.
```{r}
## Extract the foldchanges
foldchanges <- res_tableOE_tb$log2FoldChange

## Name each fold change with the gene name
names(foldchanges) <- res_tableOE_tb$gene

## Sort fold changes in decreasing order
foldchanges <- sort(foldchanges, decreasing = TRUE)
```


We can explore the enrichment of BP Gene Ontology terms using gene set enrichment analysis (GSEA) using gene sets associated with BP Gene Ontology terms
```{r}
gseaGO <- clusterProfiler::gseGO(
  geneList = foldchanges,
  ont = "BP",
  keyType = "SYMBOL",
  eps = 0,
  minGSSize = 20,
  maxGSSize = 300,
  pAdjustMethod = "BH",
  pvalueCutoff = 0.05,
  verbose = TRUE,
  OrgDb = "org.Hs.eg.db",
  by = "fgsea"
)

gseaGO_results <- data.frame(gseaGO)

```


### Visualizing GSEA Results

Visualize GSEA results with `dotplot()` for significant categories and `gseaplot2()` for gene set ranks.

```{r}
dotplot(gseaGO, showCategory = 30)
gseaplot2(gseaGO, geneSetID = 1:4)
```

The emapplot clusters the 50 most significant (by padj) GO terms to visualize relationships between terms.
```{r}

# find the overlap between the top 50 terms
pwt <- pairwise_termsim(
  gseaGO,
  method = "JC",
  semData = NULL
  )

emapplot(pwt, showCategory = 30, node_label = "all", 
         cex_label_group = 3, 
         layout.params = list("mds")) + 
         theme(legend.position = "none")



```

### GSEA using fgsea
GO annotations are updated regularly, and the GO terms associated with genes can change over time. This can lead to inconsistencies in the results of GO enrichment analyses when using different versions of the GO annotations. To address this issue, the `GOenrichment` package provides `hGOBP.gmt` a recently downloaded version of `BP` GO annotations. This file is a list of GO terms and their associated genes, which is used as input for the enrichment analysis. 

To run GSEA using the `fgsea` package, we need to load the gene sets from the `hGOBP.gmt` file and the gene-level statistics from the `foldchanges` vector. We can then run the GSEA analysis using the `fgseaSimple` function.
```{r}

# Let's look at the hGOBP.gmt file
hGOBP.gmt[1]

fgseaRes <-  fgsea::fgseaSimple(pathways = hGOBP.gmt, 
                                stats = foldchanges, 
                                nperm = 1000, maxSize = 300, 
                                minSize = 20)

fgsea <- data.frame(fgseaRes,stringsAsFactors = F)

w = which(fgsea$ES > 0)

fposgsea <- fgsea[w,]

fposgsea <- fposgsea %>% arrange(padj)

```
The `fgseaSimple` function returns a list of enriched gene sets and the enrichment score (ES).

We are going to compare these results to running the GO enrichment function `runGORESP`. 

Let's first load the `GOenrichment` package and check the available functions:
```{r}

library(GOenrichment)

ls("package:GOenrichment")
```

### `GOenrichment` Package Analysis

`runGORESP` uses over-representation analysis to identify enriched GO terms and returns two `data.frames`; of enriched GO terms (nodes) and GO term relationships (edges). 

`compSCORE`evaluates a matrix of fitness scores to identify the `querySet` with scores above a specified significance threshold.

We'll use a significance cutoff of 0.58, corresponding to a 1.5x change in expression.
```{r}
args(runGORESP)

# ?runGORESP

# Define the query set
matx <- cbind(foldchanges,foldchanges)

scoreMat = compSCORE(matx, coln = 1, sig =  0.58)

head(scoreMat)

hresp = runGORESP(fdrThresh = 0.2, scoreMat = scoreMat,
  bp_input = hGOBP.gmt, go_input = NULL, minSetSize = 20,
  maxSetSize = 300)

names(hresp$edgeMat)
names(hresp$enrichInfo)

head(hresp$enrichInfo[,c(2,3,4,5,10)])

```

Let's check the overlap between the enriched terms found using `runGORESP` and those found using `fgseaSimple` as they used the same GO term libraries:
```{r}

w = which(fposgsea$padj <= 0.2)

lens <- length(intersect(fposgsea$pathway[w],hresp$enrichInfo$term))

length(w)

dim(hresp$enrichInfo)

percent_overlap <- lens/nrow(hresp$enrichInfo)*100

percent_overlap
```

80%, that's very good because we are using two different GO enrichment methods, over-representation analysis and GSEA.

The `visNetwork` function allows us to create interactive network visualizations where enriched GO terms are nodes and the relationships between terms are edges.

The equivalent function in the `GOenrichment` package is `runNetwork`:

We use the`visSetup` function to prepare the data for network visualization. We then run the `runNetwork` function to generate the interactive network plot.
```{r, echo=FALSE, results='asis'}

vis = visSetup(hresp$enrichInfo,hresp$edgeMat)
GOenrichment::runNetwork(vis$nodes, vis$edges, main = "MOV10 Overexpression GO Network")

```

This network analysis is based on [Cytoscape](https://cytoscape.org/), an open source bioinformatics software platform for visualizing molecular interaction networks. out of all the GO packages.

This network analysis is based on [Cytoscape](https://cytoscape.org/), an open source bioinformatics software platform for visualizing molecular interaction networks. out of all the GO packages.

**Exercise** points = +4/4

1. Perform a GO enrichment analysis using the `runGORESP` function from the `GOenrichment` package using the res_tableKD_tb$log2FoldChange and cbind to build an input matrix. Use a significance threshold of 0.58 and an FDR threshold of 0.2. Save the results in a variable called `kresp`.

```{r}

kdfoldchanges <- res_tableKD_tb$log2FoldChange

names(kdfoldchanges) <- res_tableKD_tb$gene

kdfoldchanges <- sort(kdfoldchanges, decreasing = TRUE)

matkd <- cbind(kdfoldchanges,kdfoldchanges)

scoreMatkd = compSCORE(matkd, coln = 1, sig =  0.58)

kresp = runGORESP(fdrThresh = 0.2,bp_input = hGOBP.gmt,go_input = NULL,minSetSize = 20,maxSetSize = 300,scoreMat = scoreMatkd)

```

2. Visualize the enriched GO terms using the `runNetwork` function from the `GOenrichment` package. Save the results in a variable called `kvis`.
```{r}

kvis = visSetup(kresp$enrichInfo,kresp$edgeMat)

runNetwork(kvis$nodes,kvis$edges)

```

**Homework: fill in the missing code for the DGE_analysis_exercises.Rmd in the exercises folder.** points = +20.
