---
output: html_document
editor_options:
  chunk_output_type: console
---
# Functional analysis of RNAseq data

### Ontology and GO Terms

**Ontology**: 
An ontology is a representation of something we know about. Ontologies consist of representations of things that are detectable or directly observable and the relationships between those things. 
The **Gene Ontology (GO)** project provides an ontology of defined terms representing gene product properties. These terms are connected to each other via formally defined relationships, allowing for consistent annotation and analysis of gene functions across different species and research areas.

### GO Terms (Gene Ontology Terms)

The GO terms describe three aspects of gene products:

- **Biological Process**: Refers to the biological objectives achieved by gene products, such as “cell division” or “response to stress.”

![Alt text](img/go_heirarchy.jpg)

- **Molecular Function**: Describes the biochemical activities at the molecular level, such as “enzyme activity” or “DNA binding.”

- **Cellular Component**: Indicates where gene products are located within the cell, like “nucleus” or “mitochondrion.”

### GO Enrichment Methods

![Alt text](img/pathway_analysis.png)

GO enrichment methods help interpret differentially expressed gene lists from RNA-Seq data and include the following:

1. **Overrepresentation Analysis (ORA)**:
- ORA assesses whether specific GO terms are over-represented in a gene list compared to the background set of genes (e.g., all genes in the genome). Over-represented GO terms indicate biological functions or processes that are significantly associated with the genes of interest.


![Alt text](img/go_proportions.png){ width=600 }


![Alt text](img/go_proportions_table3.png){ width=600 }`

- **Hypergeometric Test**: 
The hypergeometric test in GO enrichment analysis is used to determine whether a set of genes is enriched for specific GO terms compared to the background set of genes. The test calculates the probability of observing the overlap between the query gene set and the gene set associated with a particular GO term, given the total number of genes in the dataset and the number of genes in the background set.

The formula for the cumulative probability of observing **at most \( q \)** successes (the lower-tail probability) in a sample of size \( k \) from a population with \( m \) successes and \( N - m \) failures is:
$$
P(X \leq q) = \sum_{i=0}^{q} \frac{\binom{m}{i} \binom{N - m}{k - i}}{\binom{N}{k}}
$$
Where:

**`q`**: The **number of genes in your list of interest that are annotated with a specific GO term** (e.g., genes in your differential expression results that are associated with "cell cycle regulation").

**`m`**: The **total number of genes in the background set (e.g., all genes in the genome) that are annotated with the specific GO term** (e.g., all genes associated with "cell cycle regulation" in the entire dataset or genome).

**`N`**: The **total number of genes in the background set.**

**`k`**: The **total number of genes in your gene list of interest** 

### Example
Suppose:your background population has \( N = 20,000 \) genes (e.g., the entire genome). Out of these, \( m = 500 \) genes are associated with a specific GO term (e.g., "cell cycle regulation"). You have a list of \( k = 1000 \) genes of interest. In this list, \( q = 30 \) genes are associated with the "cell cycle regulation" GO term.
```{r}
# Define parameters
N <- 20000         # Total genes in the background (population)
m <- 500           # Total genes associated with the GO term in the background
k <- 1000          # Sample size (genes in your list)
q <- 50            # Observed number of successes (overlap with GO term)

# Perform the hypergeometric test (upper-tail probability)
p_value <- phyper(q - 1, m, N - m, k, lower.tail = FALSE)
p_value
```

2. **Gene Set Enrichment Analysis (GSEA)**: 
Instead of only considering genes above a certain threshold (e.g., differentially expressed genes), GSEA looks at the **entire ranked list of genes** (e.g., ranked by their expression levels) and checks whether genes associated with a certain GO term are *statistically over-represented at the top or bottom of a ranked gene list, indicating coordinated changes in gene expression*.

Here’s how GSEA works:
   
- **Gene Ranking**:
Genes are ranked based on a metric that reflects their differential expression, such as log2FoldChange.
   
- **Enrichment Score (ES)**: 
GSEA calculates an **Enrichment Score (ES)** for each gene set by walking down the ranked list, increasing the score when a gene belongs to the set and decreasing it otherwise. A positive ES suggests upregulation, while a negative ES suggests downregulation of the gene set.
   
- **Example:**
The GSEA enrichment plot provides a graphical view of the enrichment score for a gene set:

![Alt text](img/GSEA.jpg)
- The top portion of the plot shows the running ES for the gene set as the analysis walks down the ranked list. The score at the peak of the plot (the score furthest from 0.0) is the ES for the gene set. 

- The middle portion of the plot shows where the members of the gene set appear in the ranked list of genes.

- The leading edge subset of a gene set is the subset of members that contribute most to the ES. 

- The bottom portion of the plot shows the value of the ranking metric (such as log2FoldChange) as you move down the list of ranked genes. 


### Over-representation Analysis (ORA) using `clusterProfiler`
```{r, message=FALSE}


# Uncomment the following if you haven't yet installed the required packages
# BiocManager::install(c("ggraph", "igraph", "visNetwork", 
# "GO.db", "GOSemSim"), force = TRUE)

# IMPORTANT!!! Uncomment the following to update the GOenrichment package
# remove.package("GOenrichment")
# devtools::install_github("gurinina/GOenrichment", force = TRUE)

## Load required libraries
library(org.Hs.eg.db)#
library(clusterProfiler)#
library(tidyverse)
library(enrichplot)#
library(fgsea) #
library(igraph)
library(ggraph)
library(visNetwork)
library(GO.db)
library(GOSemSim)
library(GOenrichment)
library(tidyr)
```

#### Running clusterProfiler: we first need to load the results of the differential expression analysis.
```{r}

res_tableOE = readRDS("data/res_tableOE.RDS")

res_tableOE_tb <- res_tableOE %>% 
  data.frame() %>% rownames_to_column(var = "gene") %>% 
  dplyr::filter(!is.na(log2FoldChange))  %>% as_tibble()

```

To perform the over-representation analysis, we need a list of background genes and a list of significant genes:
```{r}

## background set of genes
allOE_genes <- res_tableOE_tb$gene


sigOE = dplyr::filter(res_tableOE_tb, padj < 0.05)

## significant genes
sigOE_genes = sigOE$gene

```

The`enrichGO()` function performs the ORA for the significant genes of interest (`sigOE_genes`) compared to the background gene list (`allOE_genes`) and returns the enriched GO terms and their associated p-values. 
```{r}

## Run GO enrichment analysis
ego <- enrichGO(gene = sigOE_genes,
 universe = allOE_genes,
 keyType = "SYMBOL",
 OrgDb = org.Hs.eg.db,
 minGSSize = 20,
 maxGSSize = 300,
 ont = "BP",
 pAdjustMethod = "BH",
 qvalueCutoff = 0.05,
 readable = TRUE)

```

We can extract the gene lists associated with each enriched GO term from the `ego` object to create edges between terms based on shared genes.
```{r}
# Extract the result as a data frame
enrich_df <- as.data.frame(ego)

names(enrich_df)
head(enrich_df)

# GO term IDs 
term_ids <- enrich_df$ID  

# Split gene lists by "/" and assign term IDs as names
gene_lists <- strsplit(enrich_df$geneID, "/") 

# Assign term IDs as names
names(gene_lists) <- term_ids  

```

Using the gene lists associated with each GO term, we can create edges between terms based on the shared genes.
```{r}

# Load GO data for semantic similarity calculation
go_data <- GOSemSim::godata(OrgDb = "org.Hs.eg.db", 
                            ont = "BP", 
                            keytype = "SYMBOL",
                            computeIC = TRUE)
            

# Compute semantic similarity
similarity_matrix <- GOSemSim::mgoSim(term_ids, term_ids, 
                     semData = go_data, 
                     measure = "Wang", combine = NULL)

# Convert similarity matrix to edges
edges <- data.frame(from = rep(term_ids, 
                    each = length(term_ids)),
                    to = rep(term_ids, 
                    times = length(term_ids)),
                    similarity = as.vector(similarity_matrix))

# Filter out edges with zero similarity or self-loops
edges <- subset(edges, similarity > 0 & from != to)

edges <- edges %>% 
  dplyr::filter(similarity >= 0.5)


# Convert term IDs to names
edges$to <- Term(edges$to)
edges$from <- Term(edges$from)

head(edges)

```

Now we can create a network plot to visualize the relationships between the enriched GO terms based on their semantic similarity.
```{r}
# Create an igraph object from the edges data frame
g <- graph_from_data_frame(edges, directed = FALSE)

# Visualize the graph using ggraph
ggraph(g, layout = 'fr') +  # Fruchterman-Reingold layout
  geom_edge_link(aes(edge_alpha = 0.5)) +  # Edges with some transparency
  geom_node_point(size = 5, color = "blue") +  # Nodes sized uniformly
  geom_node_text(aes(label = name), repel = TRUE, size = 3) +  # Node labels
  theme_void() +  # Clean theme without axes
  labs(title = "GO Terms Semantic Similarity Network")  # Title for the plot

```

Edges based can also be defined by Jaccard similarity; defined as the size of the intersection each pair of GO terms divided by the size of their union 
```{r}

# Define a function to compute Jaccard similarity
jaccard_index <- function(genes1, genes2) {
length(intersect(genes1, genes2)) / length(union(genes1, genes2))
}

# Create an empty data frame to store edges
edges <- data.frame(from = character(), to = character(), 
                    similarity = numeric())
  # Compute Jaccard similarity for each pair of terms
  for (i in 1:(length(term_ids) - 1)) {
  for (j in (i + 1):length(term_ids)) {
  similarity <- jaccard_index(gene_lists[[i]], gene_lists[[j]])
  if (similarity > 0) {  # Only keep edges with some overlap
  edges <- rbind(edges, data.frame(from = term_ids[i], 
                                   to = term_ids[j], 
                                   similarity = similarity))
  }}}

edges <- edges %>% 
  dplyr::filter(similarity >= 0.5)

# Convert term IDs to names
edges$to <- Term(edges$to)
edges$from <- Term(edges$from)

```

Now we can create a network plot to visualize the relationships between the enriched GO terms based on the Jaccard similarity of their gene lists.
```{r}
# Create a graph from the edges
g <- graph_from_data_frame(edges, directed = FALSE)

# Visualize the graph using ggraph
ggraph(g, layout = 'fr') +  # Fruchterman-Reingold layout
  geom_edge_link(aes(edge_alpha = 0.5)) +  # Edges with some transparency
  geom_node_point(size = 5, 
                  color = "blue") + 
  geom_node_text(aes(label = name), 
                 repel = TRUE, 
                 size = 3) +  # Node labels
  theme_void() +  # Clean theme without axes
  labs(title = "GO Terms Jaccard Overlap Network") 

```

The resulting network plots shows the relationships between the enriched GO terms based on either Semantic or Jaccard similarity of their gene lists. 

### Visualizing Enrichment Results

**Network Plot**:
The `emapplot()` function generates a network plot where each node represents an enriched gene set, and edges between nodes indicate the similarity or overlap between those gene sets. 
```{r}

pwt <- pairwise_termsim(
  ego,
  method = "JC",
  semData = NULL
)

# category labels
emapplot(pwt, showCategory = 30, 
         node_label = "category", 
         cex_label_group = 3, 
         layout.params = list("mds")) +
         theme(legend.position = "none")

# group labels 
emapplot(pwt, showCategory = 150, 
         node_label = "group") +
         theme(legend.position = "none")

```
To save any of these figures if they become too cluttered, click on the Export button in the RStudio Plots tab and Save as PDF. In the pop-up window, change the PDF size to 24 x 32 to give a figure of appropriate size for the text labels.

**cnetplot**: Visualizes connections between genes and top GO terms, highlighting shared pathways.This plot is particularly useful for hypothesis generation in identifying genes that may be important to several of the most affected processes.
```{r}

## Extract the foldchanges
OE_foldchanges <- sigOE$log2FoldChange

names(OE_foldchanges) <- sigOE$gene

## cnetplot for the top 5 categories
cnetplot(ego,
  categorySize="padj",
  node_label = "all",
  showCategory = 5,
  foldChange = OE_foldchanges,
  vertex.label.font = 6)

```

### GSEA Using clusterProfiler and fgsea

Prepare fold changes, sort by expression, and run GSEA with `gseGO()` for clusterProfiler or `fgseaSimple()` for `fgsea`.
```{r}
## Extract the foldchanges
foldchanges <- res_tableOE_tb$log2FoldChange

## Name each fold change with the gene name
names(foldchanges) <- res_tableOE_tb$gene

## Sort fold changes in decreasing order
foldchanges <- sort(foldchanges, decreasing = TRUE)
```


We can explore the enrichment of BP Gene Ontology terms using gene set enrichment analysis (GSEA) using gene sets associated with BP Gene Ontology terms
```{r}
gseaGO <- clusterProfiler::gseGO(
  geneList = foldchanges,
  ont = "BP",
  keyType = "SYMBOL",
  eps = 0,
  minGSSize = 20,
  maxGSSize = 300,
  pAdjustMethod = "BH",
  pvalueCutoff = 0.05,
  verbose = TRUE,
  OrgDb = "org.Hs.eg.db",
  by = "fgsea"
)

gseaGO_results <- data.frame(gseaGO)

```


### Visualizing GSEA Results

Visualize GSEA results with `dotplot()` for significant categories and `gseaplot2()` for gene set ranks.

```{r}
dotplot(gseaGO, showCategory = 30)
gseaplot2(gseaGO, geneSetID = 1:4)
```

The emapplot clusters the 50 most significant (by padj) GO terms to visualize relationships between terms.
```{r}

# find the overlap between the top 50 terms
pwt <- pairwise_termsim(
  gseaGO,
  method = "JC",
  semData = NULL
  )

emapplot(pwt, showCategory = 30, node_label = "all", 
         cex_label_group = 3, 
         layout.params = list("mds")) + 
         theme(legend.position = "none")



```

### GSEA using fgsea
GO annotations are updated regularly, and the GO terms associated with genes can change over time. This can lead to inconsistencies in the results of GO enrichment analyses when using different versions of the GO annotations. To address this issue, the `GOenrichment` package provides `hGOBP.gmt` a recently downloaded version of `BP` GO annotations. This file is a list of GO terms and their associated genes, which is used as input for the enrichment analysis. 

To run GSEA using the `fgsea` package, we need to load the gene sets from the `hGOBP.gmt` file and the gene-level statistics from the `foldchanges` vector. We can then run the GSEA analysis using the `fgseaSimple` function.
```{r}


# Let's look at the hGOBP.gmt file
hGOBP.gmt[1]

fgseaRes <-  fgsea::fgseaSimple(pathways = hGOBP.gmt, 
                                stats = foldchanges, 
                                nperm = 1000, maxSize = 300, 
                                minSize = 20)

fgsea <- data.frame(fgseaRes,stringsAsFactors = F)

w = which(fgsea$ES > 0)

fposgsea <- fgsea[w,]

fposgsea <- fposgsea %>% arrange(padj)

plotEnrichment(hGOBP.gmt[["EXTRACELLULAR MATRIX ORGANIZATION"]], foldchanges)

```
The `fgseaSimple` function returns a list of enriched gene sets and the enrichment score (ES).

We are going to compare these results to running the GO enrichment function `runGORESP`. 

Let's first load the `GOenrichment` package and check the available functions:
```{r}

# Uncomment the following if you haven't yet installed GOenrichment.

# devtools::install_github("gurinina/GOenrichment")

library(GOenrichment)

ls("package:GOenrichment")
```

### `GOenrichment` Package Analysis

`runGORESP` uses over-representation analysis to identify enriched GO terms and returns two `data.frames`; of enriched GO terms (nodes) and GO term relationships (edges). 

`compSCORE`evaluates a matrix of fitness scores to identify the `querySet` with scores above a specified significance threshold.

We'll use a significance cutoff of 0.58, corresponding to a 1.5x change in expression.
```{r}
args(runGORESP)

# ?runGORESP

# Define the query set
matx <- cbind(foldchanges,foldchanges)

scoreMat = compSCORE(matx, coln = 1, sig =  0.58)

head(scoreMat)

hresp = runGORESP(fdrThresh = 0.2, scoreMat = scoreMat,
  bp_input = hGOBP.gmt, go_input = NULL, minSetSize = 20,
  maxSetSize = 300)

names(hresp$edgeMat)
names(hresp$enrichInfo)

head(hresp$enrichInfo[,c(2,3,4,5,10)])

```

Let's check the overlap between the enriched terms found using `runGORESP` and those found using `fgseaSimple` as they used the same GO term libraries:
```{r}

w = which(fposgsea$padj <= 0.2)

lens <- length(intersect(fposgsea$pathway[w],hresp$enrichInfo$term))

length(w)

dim(hresp$enrichInfo)

percent_overlap <- lens/nrow(hresp$enrichInfo)*100

percent_overlap
```

80%, that's very good because we are using two different GO enrichment methods, over-representation analysis and GSEA.

We can visualize the results of the GO enrichment analysis using the `visNetwork` package. This package allows us to create interactive network visualizations.

The equivalent function in the `GOenrichment` package is `runNetwork`:

We use the`visSetup` function to prepare the data for network visualization. We then run the `runNetwork` function to generate the interactive network plot.

```{r, echo=TRUE, results='asis'}

vis = visSetup(hresp$enrichInfo,hresp$edgeMat)
GOenrichment::runNetwork(vis$nodes, vis$edges, main = "MOV10 Overexpression GO Enrichment")

```

This network analysis is based on [Cytoscape](https://cytoscape.org/), an open source bioinformatics software platform for visualizing molecular interaction networks. out of all the GO packages.

**Exercise** points = +4

1. Perform a GO enrichment analysis using the `runGORESP` function from the `GOenrichment` package using the res_tableKD_tb$log2FoldChange and cbind to build an input matrix. Use a significance threshold of 0.58 and an FDR threshold of 0.2. Save the results in a variable called `kresp`.

2. Visualize the enriched GO terms using the `runNetwork` function from the `GOenrichment` package. Save the results in a variable called `kvis`.

### Other tools and resources

- [GeneMANIA](http://genemania.org/). GeneMANIA finds other genes that are related to a set of input genes, using a very large set of functional association data curated from the literature. 

- [ReviGO](http://revigo.irb.hr/). Revigo is an online GO enrichment tool that allows you to copy-paste your significant gene list and your background gene list. The output is a visualization of enriched GO terms in a hierarchical tree.

- [AmiGO](http://amigo.geneontology.org/amigo). AmiGO is the current official web-based set of tools for searching and browsing the Gene Ontology database.

- etc.

