<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 1 Differential gene expression (DGE) analysis overview | 2024_RNA-seq-analysis</title>
  <meta name="description" content="Chapter 1 Differential gene expression (DGE) analysis overview | 2024_RNA-seq-analysis" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 1 Differential gene expression (DGE) analysis overview | 2024_RNA-seq-analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="https://bookdown.org/ggiaever/2024_RNA-seq-analysis/img/football.png" />
  
  <meta name="github-repo" content="gurinina/2024_RNA-seq-analysis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 1 Differential gene expression (DGE) analysis overview | 2024_RNA-seq-analysis" />
  
  
  <meta name="twitter:image" content="https://bookdown.org/ggiaever/2024_RNA-seq-analysis/img/football.png" />




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="count-normalization.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/vis-9.1.0/vis-network.min.css" rel="stylesheet" />
<script src="libs/vis-9.1.0/vis-network.min.js"></script>
<script src="libs/visNetwork-binding-2.1.2/visNetwork.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="css/git.css" type="text/css" />
<link rel="stylesheet" href="css/toc.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>RNA-seq analysis workshop</a></li>
<li class="chapter" data-level="1" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html"><i class="fa fa-check"></i><b>1</b> Differential gene expression (DGE) analysis overview</a>
<ul>
<li class="chapter" data-level="1.1" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#review-of-the-dataset"><i class="fa fa-check"></i><b>1.1</b> Review of the dataset</a></li>
<li class="chapter" data-level="1.2" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#setting-up"><i class="fa fa-check"></i><b>1.2</b> Setting up</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#loading-libraries"><i class="fa fa-check"></i><b>1.2.1</b> Loading libraries</a></li>
<li class="chapter" data-level="1.2.2" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#loading-data"><i class="fa fa-check"></i><b>1.2.2</b> Loading data</a></li>
<li class="chapter" data-level="1.2.3" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#viewing-data"><i class="fa fa-check"></i><b>1.2.3</b> Viewing data</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#differential-gene-expression-analysis-overview"><i class="fa fa-check"></i><b>1.3</b> Differential gene expression analysis overview</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#rna-seq-count-distribution"><i class="fa fa-check"></i><b>1.3.1</b> RNA-seq count distribution</a></li>
<li class="chapter" data-level="1.3.2" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#modeling-count-data"><i class="fa fa-check"></i><b>1.3.2</b> Modeling count data</a></li>
<li class="chapter" data-level="1.3.3" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#improving-mean-estimates-i.e.-reducing-variance-with-biological-replicates"><i class="fa fa-check"></i><b>1.3.3</b> Improving mean estimates (i.e. reducing variance) with biological replicates</a></li>
<li class="chapter" data-level="1.3.4" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#differential-expression-analysis-workflow"><i class="fa fa-check"></i><b>1.3.4</b> Differential expression analysis workflow</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="count-normalization.html"><a href="count-normalization.html"><i class="fa fa-check"></i><b>2</b> Count normalization</a>
<ul>
<li class="chapter" data-level="2.1" data-path="count-normalization.html"><a href="count-normalization.html#normalization"><i class="fa fa-check"></i><b>2.1</b> Normalization</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="count-normalization.html"><a href="count-normalization.html#common-normalization-methods"><i class="fa fa-check"></i><b>2.1.1</b> Common normalization methods</a></li>
<li class="chapter" data-level="2.1.2" data-path="count-normalization.html"><a href="count-normalization.html#tpm-not-recommended"><i class="fa fa-check"></i><b>2.1.2</b> TPM (Not Recommended)</a></li>
<li class="chapter" data-level="2.1.3" data-path="count-normalization.html"><a href="count-normalization.html#rpkmfpkm-not-recommended"><i class="fa fa-check"></i><b>2.1.3</b> RPKM/FPKM (Not Recommended)</a></li>
<li class="chapter" data-level="2.1.4" data-path="count-normalization.html"><a href="count-normalization.html#deseq2-normalization-median-of-ratios-method"><i class="fa fa-check"></i><b>2.1.4</b> DESeq2 Normalization: Median of Ratios Method</a></li>
<li class="chapter" data-level="2.1.5" data-path="count-normalization.html"><a href="count-normalization.html#summary-of-key-reasons-for-using-deseq2"><i class="fa fa-check"></i><b>2.1.5</b> Summary of Key Reasons for Using DESeq2</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="count-normalization.html"><a href="count-normalization.html#count-normalization-of-mov10"><i class="fa fa-check"></i><b>2.2</b> Count normalization of Mov10</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="count-normalization.html"><a href="count-normalization.html#match-the-metadata-and-counts-data"><i class="fa fa-check"></i><b>2.2.1</b> 1. Match the metadata and counts data</a></li>
<li class="chapter" data-level="2.2.2" data-path="count-normalization.html"><a href="count-normalization.html#create-deseq2-object"><i class="fa fa-check"></i><b>2.2.2</b> 2. Create DESEq2 object</a></li>
<li class="chapter" data-level="2.2.3" data-path="count-normalization.html"><a href="count-normalization.html#generate-the-mov10-normalized-counts"><i class="fa fa-check"></i><b>2.2.3</b> 3. Generate the Mov10 normalized counts</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>3</b> Quality Control</a>
<ul>
<li class="chapter" data-level="3.1" data-path="quality-control.html"><a href="quality-control.html#sample-level-qc"><i class="fa fa-check"></i><b>3.1</b> Sample-level QC</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="quality-control.html"><a href="quality-control.html#recommended-resources"><i class="fa fa-check"></i><b>3.1.1</b> Recommended Resources</a></li>
<li class="chapter" data-level="3.1.2" data-path="quality-control.html"><a href="quality-control.html#hierarchical-clustering-heatmap"><i class="fa fa-check"></i><b>3.1.2</b> Hierarchical Clustering Heatmap</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="quality-control.html"><a href="quality-control.html#gene-level-qc"><i class="fa fa-check"></i><b>3.2</b> Gene-level QC</a></li>
<li class="chapter" data-level="3.3" data-path="quality-control.html"><a href="quality-control.html#mov10-quality-assessment-and-exploratory-analysis-using-deseq2"><i class="fa fa-check"></i><b>3.3</b> Mov10 quality assessment and exploratory analysis using DESeq2</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="quality-control.html"><a href="quality-control.html#transform-normalized-counts-using-the-rlog-transformation"><i class="fa fa-check"></i><b>3.3.1</b> Transform normalized counts using the rlog transformation</a></li>
<li class="chapter" data-level="3.3.2" data-path="quality-control.html"><a href="quality-control.html#principal-components-analysis-pca"><i class="fa fa-check"></i><b>3.3.2</b> Principal components analysis (PCA)</a></li>
<li class="chapter" data-level="3.3.3" data-path="quality-control.html"><a href="quality-control.html#hierarchical-clustering"><i class="fa fa-check"></i><b>3.3.3</b> Hierarchical Clustering</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html"><i class="fa fa-check"></i><b>4</b> DGE analysis workflow</a>
<ul>
<li class="chapter" data-level="4.1" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#running-deseq2"><i class="fa fa-check"></i><b>4.1</b> Running DESeq2</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#set-the-design-formula"><i class="fa fa-check"></i><b>4.1.1</b> Set the Design Formula</a></li>
<li class="chapter" data-level="4.1.2" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#mov10-differential-expression-analysis"><i class="fa fa-check"></i><b>4.1.2</b> MOV10 Differential Expression Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#deseq2-differential-gene-expression-analysis-workflow"><i class="fa fa-check"></i><b>4.2</b> DESeq2 differential gene expression analysis workflow</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#step-1-estimate-size-factors"><i class="fa fa-check"></i><b>4.2.1</b> Step 1: Estimate size factors</a></li>
<li class="chapter" data-level="4.2.2" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#step-2-estimate-gene-wise-dispersion"><i class="fa fa-check"></i><b>4.2.2</b> Step 2: Estimate Gene-wise Dispersion</a></li>
<li class="chapter" data-level="4.2.3" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#step-3-fit-curve-to-gene-wise-dispersion-estimates"><i class="fa fa-check"></i><b>4.2.3</b> Step 3: Fit Curve to Gene-wise Dispersion Estimates</a></li>
<li class="chapter" data-level="4.2.4" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#step-4-assessing-model-fit-with-dispersion-plots"><i class="fa fa-check"></i><b>4.2.4</b> Step 4: Assessing Model Fit with Dispersion Plots</a></li>
<li class="chapter" data-level="4.2.5" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#assessing-and-addressing-issues"><i class="fa fa-check"></i><b>4.2.5</b> Assessing and Addressing Issues</a></li>
<li class="chapter" data-level="4.2.6" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#mov10-differential-expression-analysis-exploring-dispersion-estimates"><i class="fa fa-check"></i><b>4.2.6</b> MOV10 Differential Expression Analysis: Exploring Dispersion Estimates</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html"><i class="fa fa-check"></i><b>5</b> Negative Binomial model fitting</a>
<ul>
<li class="chapter" data-level="5.1" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#generalized-linear-model-fit-for-each-gene"><i class="fa fa-check"></i><b>5.1</b> Generalized Linear Model fit for each gene</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#negative-binomial-model-formula"><i class="fa fa-check"></i><b>5.1.1</b> Negative Binomial Model Formula</a></li>
<li class="chapter" data-level="5.1.2" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#estimating-beta-coefficients-using-the-design-matrix"><i class="fa fa-check"></i><b>5.1.2</b> Estimating Beta Coefficients Using the Design Matrix</a></li>
<li class="chapter" data-level="5.1.3" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#step-by-step-process"><i class="fa fa-check"></i><b>5.1.3</b> Step-by-Step Process:</a></li>
<li class="chapter" data-level="5.1.4" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#example"><i class="fa fa-check"></i><b>5.1.4</b> Example:</a></li>
<li class="chapter" data-level="5.1.5" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#calculation"><i class="fa fa-check"></i><b>5.1.5</b> Calculation:</a></li>
<li class="chapter" data-level="5.1.6" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#matrix-multiplication-form"><i class="fa fa-check"></i><b>5.1.6</b> Matrix Multiplication Form:</a></li>
<li class="chapter" data-level="5.1.7" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#log2-fold-change-and-adjustments"><i class="fa fa-check"></i><b>5.1.7</b> <strong>Log2 Fold Change and Adjustments</strong></a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#shrunken-log2-foldchanges-lfc"><i class="fa fa-check"></i><b>5.2</b> Shrunken log2 foldchanges (LFC)</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#hypothesis-testing"><i class="fa fa-check"></i><b>5.2.1</b> Hypothesis Testing</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#statistical-test-for-lfc-estimates-wald-test"><i class="fa fa-check"></i><b>5.3</b> Statistical test for LFC estimates: Wald test</a></li>
<li class="chapter" data-level="5.4" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#mov10-differential-expression-analysis-control-versus-overexpression"><i class="fa fa-check"></i><b>5.4</b> MOV10 Differential Expression Analysis: Control versus Overexpression</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#creating-contrasts-for-hypothesis-testing"><i class="fa fa-check"></i><b>5.4.1</b> Creating Contrasts for Hypothesis Testing</a></li>
<li class="chapter" data-level="5.4.2" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#ma-plot"><i class="fa fa-check"></i><b>5.4.2</b> MA Plot</a></li>
<li class="chapter" data-level="5.4.3" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#multiple-testing-correction"><i class="fa fa-check"></i><b>5.4.3</b> Multiple Testing Correction</a></li>
<li class="chapter" data-level="5.4.4" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#benjamini-hochberg-adjustment"><i class="fa fa-check"></i><b>5.4.4</b> Benjamini-Hochberg Adjustment</a></li>
<li class="chapter" data-level="5.4.5" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#mov10-de-analysis-control-vs.-knockdown"><i class="fa fa-check"></i><b>5.4.5</b> MOV10 DE Analysis: <strong>Control vs. Knockdown</strong></a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#summarizing-results"><i class="fa fa-check"></i><b>5.5</b> Summarizing Results</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#extracting-significant-differentially-expressed-genes"><i class="fa fa-check"></i><b>5.5.1</b> Extracting Significant Differentially Expressed Genes</a></li>
<li class="chapter" data-level="5.5.2" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#mov10-knockdown-analysis-control-vs.-knockdown"><i class="fa fa-check"></i><b>5.5.2</b> MOV10 Knockdown Analysis: Control vs. Knockdown</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="visualizing-rna-seq-results.html"><a href="visualizing-rna-seq-results.html"><i class="fa fa-check"></i><b>6</b> Visualizing RNA-seq results</a>
<ul>
<li class="chapter" data-level="6.0.1" data-path="visualizing-rna-seq-results.html"><a href="visualizing-rna-seq-results.html#plotting-signicant-de-genes"><i class="fa fa-check"></i><b>6.0.1</b> Plotting signicant DE genes</a></li>
<li class="chapter" data-level="6.0.2" data-path="visualizing-rna-seq-results.html"><a href="visualizing-rna-seq-results.html#heatmap"><i class="fa fa-check"></i><b>6.0.2</b> Heatmap</a></li>
<li class="chapter" data-level="6.0.3" data-path="visualizing-rna-seq-results.html"><a href="visualizing-rna-seq-results.html#volcano-plots"><i class="fa fa-check"></i><b>6.0.3</b> Volcano plots</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="summary-of-dge-workflow.html"><a href="summary-of-dge-workflow.html"><i class="fa fa-check"></i><b>7</b> Summary of DGE workflow</a>
<ul>
<li class="chapter" data-level="7.0.1" data-path="summary-of-dge-workflow.html"><a href="summary-of-dge-workflow.html#import-data-into-dds-object"><i class="fa fa-check"></i><b>7.0.1</b> 1. Import data into dds object:</a></li>
<li class="chapter" data-level="7.0.2" data-path="summary-of-dge-workflow.html"><a href="summary-of-dge-workflow.html#exploratory-data-analysis-pca-heirarchical-clustering---identifying-outliers-and-sources-of-variation-in-the-data"><i class="fa fa-check"></i><b>7.0.2</b> 2. Exploratory data analysis (PCA &amp; heirarchical clustering) - identifying outliers and sources of variation in the data:</a></li>
<li class="chapter" data-level="7.0.3" data-path="summary-of-dge-workflow.html"><a href="summary-of-dge-workflow.html#run-deseq2"><i class="fa fa-check"></i><b>7.0.3</b> 3. Run DESeq2:</a></li>
<li class="chapter" data-level="7.0.4" data-path="summary-of-dge-workflow.html"><a href="summary-of-dge-workflow.html#check-the-fit-of-the-dispersion-estimates"><i class="fa fa-check"></i><b>7.0.4</b> 4. Check the fit of the dispersion estimates:</a></li>
<li class="chapter" data-level="7.0.5" data-path="summary-of-dge-workflow.html"><a href="summary-of-dge-workflow.html#create-contrasts-to-perform-wald-testing-on-the-shrunken-log2-foldchanges-between-specific-conditions"><i class="fa fa-check"></i><b>7.0.5</b> 5. Create contrasts to perform Wald testing on the shrunken log2 foldchanges between specific conditions:</a></li>
<li class="chapter" data-level="7.0.6" data-path="summary-of-dge-workflow.html"><a href="summary-of-dge-workflow.html#output-significant-results"><i class="fa fa-check"></i><b>7.0.6</b> 6. Output significant results:</a></li>
<li class="chapter" data-level="7.0.7" data-path="summary-of-dge-workflow.html"><a href="summary-of-dge-workflow.html#visualize-results-volcano-plots-heatmaps-normalized-counts-plots-of-top-genes-etc."><i class="fa fa-check"></i><b>7.0.7</b> 7. Visualize results: volcano plots, heatmaps, normalized counts plots of top genes, etc.</a></li>
<li class="chapter" data-level="7.0.8" data-path="summary-of-dge-workflow.html"><a href="summary-of-dge-workflow.html#make-sure-to-output-the-versions-of-all-tools-used-in-the-de-analysis"><i class="fa fa-check"></i><b>7.0.8</b> 8. Make sure to output the versions of all tools used in the DE analysis:</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html"><i class="fa fa-check"></i><b>8</b> Functional analysis of RNAseq data</a>
<ul>
<li class="chapter" data-level="8.0.1" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#ontology-and-go-terms"><i class="fa fa-check"></i><b>8.0.1</b> Ontology and GO Terms</a></li>
<li class="chapter" data-level="8.0.2" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#go-terms-gene-ontology-terms"><i class="fa fa-check"></i><b>8.0.2</b> GO Terms (Gene Ontology Terms)</a></li>
<li class="chapter" data-level="8.0.3" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#go-enrichment-methods"><i class="fa fa-check"></i><b>8.0.3</b> GO Enrichment Methods</a></li>
<li class="chapter" data-level="8.0.4" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#example-1"><i class="fa fa-check"></i><b>8.0.4</b> Example</a></li>
<li class="chapter" data-level="8.0.5" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#over-representation-analysis-ora-using-clusterprofiler"><i class="fa fa-check"></i><b>8.0.5</b> Over-representation Analysis (ORA) using <code>clusterProfiler</code></a></li>
<li class="chapter" data-level="8.0.6" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#visualizing-enrichment-results"><i class="fa fa-check"></i><b>8.0.6</b> Visualizing Enrichment Results</a></li>
<li class="chapter" data-level="8.0.7" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#gsea-using-clusterprofiler-and-fgsea"><i class="fa fa-check"></i><b>8.0.7</b> GSEA Using clusterProfiler and fgsea</a></li>
<li class="chapter" data-level="8.0.8" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#visualizing-gsea-results"><i class="fa fa-check"></i><b>8.0.8</b> Visualizing GSEA Results</a></li>
<li class="chapter" data-level="8.0.9" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#gsea-using-fgsea"><i class="fa fa-check"></i><b>8.0.9</b> GSEA using fgsea</a></li>
<li class="chapter" data-level="8.0.10" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#goenrichment-package-analysis"><i class="fa fa-check"></i><b>8.0.10</b> <code>GOenrichment</code> Package Analysis</a></li>
<li class="chapter" data-level="8.0.11" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#other-tools-and-resources"><i class="fa fa-check"></i><b>8.0.11</b> Other tools and resources</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">2024_RNA-seq-analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="differential-gene-expression-dge-analysis-overview" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">Chapter 1</span> Differential gene expression (DGE) analysis overview<a href="differential-gene-expression-dge-analysis-overview.html#differential-gene-expression-dge-analysis-overview" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The goal of RNA-seq is to perform differential expression testing to determine which genes are expressed at different levels between conditions. These genes can offer biological insight into the processes affected by the condition(s) of interest.</p>
<p>To determine the expression levels of genes, our RNA-seq workflow follows the steps detailed in the image below inside the box. All steps are performed on the command line (Linux/Unix) through the generation of read counts per gene as discussed in Corey’s lectures. The differential expression analysis and any downstream functional analysis are generally performed in R using R packages specifically designed for the complex statistical analyses required to determine whether genes are differentially expressed starting from the count matrices.</p>
<p><img src="img/rnaseq_full_workflow.png" width="400" alt="Alt text" /></p>
<p>In the next few lessons, we will walk you through an <strong>end-to-end gene-level RNA-seq differential expression workflow</strong> using various R packages. We will start with the count matrix, perform exploratory data analysis for quality assessment and to explore the relationship between samples, perform differential expression analysis, and visually explore the results prior to performing downstream functional analysis.</p>
<div id="review-of-the-dataset" class="section level2 hasAnchor" number="1.1">
<h2><span class="header-section-number">1.1</span> Review of the dataset<a href="differential-gene-expression-dge-analysis-overview.html#review-of-the-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We will be using the full count matrix from the RNA-seq dataset that is part of a larger study described in <a href="http://www.ncbi.nlm.nih.gov/pubmed/25464849">Kenny PJ et al, Cell Rep 2014</a> investigating the interactions between genes potentially involved in Fragile X syndrome (FXS). FXS is a genetic disorder and the leading cause of inherited intellectual disabilities like autism. FXS is caused by aberrant production of a protein called Fragile X Messenger Ribonucleoprotein (<strong>FMRP</strong>) that is needed for brain development. People who have FXS do not make this protein. The authors demonstrated that <strong>FMRP</strong> associates with another RNA-binding protein <strong>MOV10</strong> (Mov10 RISC Complex RNA Helicase) and acts to regulate the translation of a subset of RNAs.</p>
<p>What is the function of <strong>FMRP</strong> and <strong>MOV10</strong>?</p>
<blockquote>
<p><strong>FMRP</strong> is “most commonly found in the brain, is essential for normal cognitive development and female reproductive function. Mutations of this gene can lead to fragile X syndrome, mental retardation, premature ovarian failure, autism, Parkinson’s disease, developmental delays and other cognitive deficits.” - from <a href="https://en.wikipedia.org/wiki/FMR1">wikipedia</a></p>
</blockquote>
<blockquote>
<p><strong>MOV10</strong> MOV10 has been shown to associate with the 3’ UTR of mRNAs as part of its role in the RNA-induced silencing complex (RISC). MOV10 is an RNA helicase believed to assist in the unwinding of RNA duplexes. MOV10 is associated with <strong>FMRP</strong> in the context of the microRNA pathway. <strong>MOV10</strong> has a role in regulating translation: it facilitates microRNA-mediated translation of some RNAs, but it also increases expression of other RNAs.</p>
</blockquote>
<p>The aim of the RNAseq part of the study was to characterize the transcription expression patterns of <strong>FMRP</strong> and <strong>MOV10</strong> to identify overlapping target genes which would suggest that these genes are regulated by the <strong>MOV10-FMRP</strong> complex.</p>
<p><img src="img/model_mov10.png" width="400" alt="Alt text" /></p>
<p><strong>Model for </strong>MOV10-FMRP** Association in Translation Regulation<strong>. Top: fate of RNAs bound by </strong>MOV10<strong>. </strong>MOV10** binds the 3′ UTR-encoded G-rich structure to reveal MREs for subsequent AGO2 association. Middle: fate of RNAs bound by <strong>FMRP</strong>. <strong>FMRP</strong> binds RNAs in the nucleus. Upon export, <strong>FMRP</strong> recruits <strong>MOV10</strong>, which ultimately unwinds MREs for association with AGO2. Bottom: <strong>FMRP</strong> recruits <strong>MOV10</strong> to RNAs; however, binding of both <strong>FMRP</strong> and <strong>MOV10</strong> in proximity of MRE blocks association with AGO2. Red line indicates MRE.</p>
<p>RNA-seq was performed on HEK293F cells that were either transfected with a <strong>MOV10</strong> transgene, or siRNA to knock down Mov10 expression, or non-specific (irrelevant) siRNA. This resulted in 3 conditions <strong>Mov10 oe</strong> (over expression), <strong>Mov10 kd</strong> (knock down) and <strong>Irrelevant kd</strong>, respectively. The number of replicates is as shown below.</p>
<p><img src="img/dataset.png" width="400" alt="Alt text" /></p>
<p>Using these data, we will evaluate transcriptional patterns associated with perturbation of <strong>MOV10</strong> expression. Please note that the irrelevant siRNA will be treated as our control condition.</p>
<p><strong>Links to raw data</strong>
[GEO]: <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE51443" class="uri">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE51443</a> “Gene Expression Omnibus”
[SRA]: <a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP031507" class="uri">https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP031507</a> “Sequence Read Archive”</p>
<p><strong>Our questions:</strong></p>
<ul>
<li><p>What patterns of expression can we identify with the loss or gain of MOV10?</p></li>
<li><p>Are there any genes that are differentially expressed between the two conditions?</p></li>
<li><p>Are there any genes shared between the two conditions?</p></li>
</ul>
</div>
<div id="setting-up" class="section level2 hasAnchor" number="1.2">
<h2><span class="header-section-number">1.2</span> Setting up<a href="differential-gene-expression-dge-analysis-overview.html#setting-up" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Go to the <code>File</code> menu and to the <code>lessons</code> directory and open <code>09-DGE_codebook.Rmd</code>. This will open in the script editor in the top left hand corner. This is where we will be running all commands required for this analysis, and like last time this is where you will enter the code/answers for your homework assignment. Please save this as something with your name in it (e.g. <code>firstname_lastname_09-DGE_codebook.Rmd</code>).</p>
<p><img src="img/settingup.png" alt="setup" /></p>
<div id="loading-libraries" class="section level3 hasAnchor" number="1.2.1">
<h3><span class="header-section-number">1.2.1</span> Loading libraries<a href="differential-gene-expression-dge-analysis-overview.html#loading-libraries" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For this analysis we will be using several R packages, some which have been installed from CRAN and others from Bioconductor. To use these packages (and the functions contained within them), we need to <strong>load the libraries</strong>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="differential-gene-expression-dge-analysis-overview.html#cb1-1" tabindex="-1"></a><span class="do">## Setup</span></span>
<span id="cb1-2"><a href="differential-gene-expression-dge-analysis-overview.html#cb1-2" tabindex="-1"></a><span class="do">### Bioconductor and CRAN libraries used</span></span>
<span id="cb1-3"><a href="differential-gene-expression-dge-analysis-overview.html#cb1-3" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-4"><a href="differential-gene-expression-dge-analysis-overview.html#cb1-4" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb1-5"><a href="differential-gene-expression-dge-analysis-overview.html#cb1-5" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb1-6"><a href="differential-gene-expression-dge-analysis-overview.html#cb1-6" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb1-7"><a href="differential-gene-expression-dge-analysis-overview.html#cb1-7" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-8"><a href="differential-gene-expression-dge-analysis-overview.html#cb1-8" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span></code></pre></div>
</div>
<div id="loading-data" class="section level3 hasAnchor" number="1.2.2">
<h3><span class="header-section-number">1.2.2</span> Loading data<a href="differential-gene-expression-dge-analysis-overview.html#loading-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To load the data into our current environment, we will be using the <code>read.table</code> function. We need to provide the path to each file and also specify arguments to let R know that we have a header (<code>header = T</code>) and the first column is our row names (<code>row.names = 1</code>). By default the function expects tab-delimited files, which is what we have.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="differential-gene-expression-dge-analysis-overview.html#cb2-1" tabindex="-1"></a><span class="do">## Load in data</span></span>
<span id="cb2-2"><a href="differential-gene-expression-dge-analysis-overview.html#cb2-2" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/Mov10_full_counts.txt&quot;</span>, <span class="at">header=</span>T, <span class="at">row.names=</span><span class="dv">1</span>) </span>
<span id="cb2-3"><a href="differential-gene-expression-dge-analysis-overview.html#cb2-3" tabindex="-1"></a></span>
<span id="cb2-4"><a href="differential-gene-expression-dge-analysis-overview.html#cb2-4" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/Mov10_full_meta.txt&quot;</span>, <span class="at">header=</span>T, <span class="at">row.names=</span><span class="dv">1</span>)</span></code></pre></div>
<p>Use <code>class()</code> to inspect our data and make sure we are working with data frames:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="differential-gene-expression-dge-analysis-overview.html#cb3-1" tabindex="-1"></a><span class="do">### Check classes of the data we just brought in</span></span>
<span id="cb3-2"><a href="differential-gene-expression-dge-analysis-overview.html#cb3-2" tabindex="-1"></a><span class="fu">class</span>(meta)</span></code></pre></div>
<pre><code>## [1] &quot;data.frame&quot;</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="differential-gene-expression-dge-analysis-overview.html#cb5-1" tabindex="-1"></a><span class="fu">class</span>(data)</span></code></pre></div>
<pre><code>## [1] &quot;data.frame&quot;</code></pre>
</div>
<div id="viewing-data" class="section level3 hasAnchor" number="1.2.3">
<h3><span class="header-section-number">1.2.3</span> Viewing data<a href="differential-gene-expression-dge-analysis-overview.html#viewing-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Make sure your datasets contain the expected samples / information before proceeding to perfom any type of analysis.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="differential-gene-expression-dge-analysis-overview.html#cb7-1" tabindex="-1"></a><span class="fu">rownames</span>(meta)</span></code></pre></div>
<pre><code>## [1] &quot;Irrel_kd_1&quot; &quot;Irrel_kd_2&quot; &quot;Irrel_kd_3&quot; &quot;Mov10_kd_2&quot; &quot;Mov10_kd_3&quot; &quot;Mov10_oe_1&quot;
## [7] &quot;Mov10_oe_2&quot; &quot;Mov10_oe_3&quot;</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="differential-gene-expression-dge-analysis-overview.html#cb9-1" tabindex="-1"></a><span class="fu">names</span>(data)</span></code></pre></div>
<pre><code>## [1] &quot;Mov10_kd_2&quot; &quot;Mov10_kd_3&quot; &quot;Mov10_oe_1&quot; &quot;Mov10_oe_2&quot; &quot;Mov10_oe_3&quot; &quot;Irrel_kd_1&quot;
## [7] &quot;Irrel_kd_2&quot; &quot;Irrel_kd_3&quot;</code></pre>
<p>You’ll notice the colnames of the data are the sample names, and the rownames of the metadata are the sample names. This is important as we will be merging the metadata with the data based on the sample names.</p>
</div>
</div>
<div id="differential-gene-expression-analysis-overview" class="section level2 hasAnchor" number="1.3">
<h2><span class="header-section-number">1.3</span> Differential gene expression analysis overview<a href="differential-gene-expression-dge-analysis-overview.html#differential-gene-expression-analysis-overview" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>So what does this count data actually represent? The count data used for differential expression analysis represents the number of sequence reads that originated from a particular gene. The higher the number of counts, the more reads associated with that gene, and the assumption that there was a higher level of expression of that gene in the sample.</p>
<p><img src="img/deseq_counts_overview.png" width="600" alt="Alt text" /></p>
<p>With differential expression analysis, we are looking for genes that change in expression between two or more groups (defined in the metadata)
- case vs. control
- correlation of expression with some variable or clinical outcome</p>
<p><strong>Why does it not work to identify differentially expressed gene by ranking the genes by how different they are between the two groups (based on fold change values)?</strong></p>
<p><img src="img/foldchange_heatmap.png" width="200" alt="Alt text" /></p>
<p>More often than not, there is much more going on with your data than what you are anticipating. Genes that vary in expression level between samples is a consequence of not only the experimental variables of interest but also due to extraneous sources. The goal of differential expression analysis to determine the relative role of these effects, and to separate the “interesting” from the “uninteresting”.</p>
<p><img src="img/de_variation.png" width="500" alt="Alt text" /></p>
<p>The “uninteresting” presents as sources of variation in your data, and so even though the mean expression levels between sample groups may appear to be quite different, it is possible that the difference is not actually significant. This is illustrated for ‘GeneA’ expression between ‘untreated’ and ‘treated’ groups in the figure below. The mean expression level of geneA for the ‘treated’ group is twice as large as for the ‘untreated’ group, but the variation between replicates indicates that this may not be a significant difference. <strong>We need to take into account the variation in the data (and where it might be coming from) when determining whether genes are differentially expressed.</strong></p>
<p><img src="img/de_norm_counts_var.png" width="400" alt="Alt text" /></p>
<p>The goal of differential expression analysis is to determine, for each gene, whether the differences in expression (counts) <strong>between groups</strong> is significant given the amount of variation observed <strong>within groups</strong> (replicates). To test for significance, we need an appropriate statistical model that accurately performs normalization (to account for differences in sequencing depth, etc.) and variance modeling (to account for few numbers of replicates and large dynamic expression range).</p>
<div id="rna-seq-count-distribution" class="section level3 hasAnchor" number="1.3.1">
<h3><span class="header-section-number">1.3.1</span> RNA-seq count distribution<a href="differential-gene-expression-dge-analysis-overview.html#rna-seq-count-distribution" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To determine the appropriate statistical model, we need information about the distribution of counts. To get an idea about how RNA-seq counts are distributed, let’s plot the counts for a single sample, ‘Mov10_oe_1’:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="differential-gene-expression-dge-analysis-overview.html#cb11-1" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb11-2"><a href="differential-gene-expression-dge-analysis-overview.html#cb11-2" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x =</span> Mov10_oe_1), <span class="at">stat =</span> <span class="st">&quot;bin&quot;</span>, <span class="at">bins =</span> <span class="dv">200</span>) <span class="sc">+</span></span>
<span id="cb11-3"><a href="differential-gene-expression-dge-analysis-overview.html#cb11-3" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Raw expression counts&quot;</span>) <span class="sc">+</span></span>
<span id="cb11-4"><a href="differential-gene-expression-dge-analysis-overview.html#cb11-4" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Number of genes&quot;</span>)</span></code></pre></div>
<p><img src="2024_RNA-seq-analysis_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<p>If we zoom in close to zero, we can see a large number of genes with counts of zero:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="differential-gene-expression-dge-analysis-overview.html#cb12-1" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb12-2"><a href="differential-gene-expression-dge-analysis-overview.html#cb12-2" tabindex="-1"></a>   <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x =</span> Mov10_oe_1), <span class="at">stat =</span> <span class="st">&quot;bin&quot;</span>, <span class="at">bins =</span> <span class="dv">200</span>) <span class="sc">+</span> </span>
<span id="cb12-3"><a href="differential-gene-expression-dge-analysis-overview.html#cb12-3" tabindex="-1"></a>   <span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">500</span>)  <span class="sc">+</span></span>
<span id="cb12-4"><a href="differential-gene-expression-dge-analysis-overview.html#cb12-4" tabindex="-1"></a>   <span class="fu">xlab</span>(<span class="st">&quot;Raw expression counts&quot;</span>) <span class="sc">+</span></span>
<span id="cb12-5"><a href="differential-gene-expression-dge-analysis-overview.html#cb12-5" tabindex="-1"></a>   <span class="fu">ylab</span>(<span class="st">&quot;Number of genes&quot;</span>)</span></code></pre></div>
<p><img src="2024_RNA-seq-analysis_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<p>These images illustrate some common features of RNA-seq count data, including a <strong>low number of counts associated with a large proportion of genes</strong>, and a long right tail due to the <strong>lack of any upper limit for expression</strong>.</p>
</div>
<div id="modeling-count-data" class="section level3 hasAnchor" number="1.3.2">
<h3><span class="header-section-number">1.3.2</span> Modeling count data<a href="differential-gene-expression-dge-analysis-overview.html#modeling-count-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Count data is often modeled using the <strong>binomial distribution</strong>. The Binomial distribution is a common probability distribution that models the probability of obtaining one of two outcomes under a given number of parameters. It summarizes the number of successes in a fixed number of trials when each trial has two possible outcomes. For example, a coin toss is typically modeled using a binomial distribution, where the number of successes (e.g., heads) in a fixed number of trials (e.g., 10 coin tosses) when each trial has two possible outcomes (e.g., heads or tails) and the probability of success is constant. However, not all count data can be fit with the binomial distribution. The binomial is based on discrete events and used in situations when you have a certain number of cases.</p>
<p>The <strong>binomial distribution</strong> is used to model scenarios where there are two possible outcomes, such as “success” or “failure,” over a fixed number of trials. For example, flipping a coin 10 times and counting the number of “heads” out of those 10 flips follows a binomial distribution with a probability of success (<span class="math inline">\(p\)</span>) equal to 0.5.</p>
<p>When <strong>the number of cases is very large (i.e. people who buy lottery tickets), but the probability of an event is very small (probability of winning)</strong>, the <strong>Poisson distribution</strong> is used to model these types of count data.</p>
<p><img src="img/poisson-distribution-formula.png" width="300" alt="Alt text" /></p>
<p><strong>With RNA-seq data, for each sample we have millions of reads being sequenced and the probability of a read mapping to a gene is extremely low.</strong> Thus, it would be an appropriate situation to use the Poisson distribution. However, a unique property of this distribution is that the mean == variance given by the single parameter <span class="math inline">\(\lambda\)</span>.</p>
<p>For us to apply a poisson distribution to our data, we first need to find out
whether our data fulfills the criteria to use the poisson distribution. To do that we can plot the <em>mean versus variance</em> for the ‘Mov10 overexpression’ replicates:</p>
<p>To calculate the mean and variance of our data, we will use the <code>apply</code> function. The <code>apply</code> function allows you to apply a function to the margins (rows or columns) of a matrix. The syntax for <code>apply</code> is as follows: <code>apply(X, MARGIN, FUN)</code>, where <code>X</code> is a matrix, <code>MARGIN</code> is the margin of the matrix to apply the function to (1 = rows, 2 = columns), and <code>FUN</code> is the function to apply.</p>
<p>We will use <code>MARGIN = 1</code> to apply the function mean and variance of the counts for each row (gene) across the ‘Mov10 overexpression’ replicates. We will then create a data frame with the mean and variance of the counts for each gene.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="differential-gene-expression-dge-analysis-overview.html#cb13-1" tabindex="-1"></a><span class="co"># calculate the mean and variance of the counts for each gene</span></span>
<span id="cb13-2"><a href="differential-gene-expression-dge-analysis-overview.html#cb13-2" tabindex="-1"></a>mean_counts <span class="ot">&lt;-</span> <span class="fu">apply</span>(data[, <span class="dv">3</span><span class="sc">:</span><span class="dv">5</span>], <span class="dv">1</span>, mean)</span>
<span id="cb13-3"><a href="differential-gene-expression-dge-analysis-overview.html#cb13-3" tabindex="-1"></a></span>
<span id="cb13-4"><a href="differential-gene-expression-dge-analysis-overview.html#cb13-4" tabindex="-1"></a>variance_counts <span class="ot">&lt;-</span> <span class="fu">apply</span>(data[, <span class="dv">3</span><span class="sc">:</span><span class="dv">5</span>], <span class="dv">1</span>, var)</span>
<span id="cb13-5"><a href="differential-gene-expression-dge-analysis-overview.html#cb13-5" tabindex="-1"></a></span>
<span id="cb13-6"><a href="differential-gene-expression-dge-analysis-overview.html#cb13-6" tabindex="-1"></a><span class="co"># for ggplot we need the data to be in a data.frame</span></span>
<span id="cb13-7"><a href="differential-gene-expression-dge-analysis-overview.html#cb13-7" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(mean_counts, variance_counts)</span></code></pre></div>
<p>Run the following code to plot the <em>mean versus variance</em> for the ‘Mov10 overexpression’ replicates:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="differential-gene-expression-dge-analysis-overview.html#cb14-1" tabindex="-1"></a><span class="fu">ggplot</span>(df) <span class="sc">+</span></span>
<span id="cb14-2"><a href="differential-gene-expression-dge-analysis-overview.html#cb14-2" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> mean_counts, <span class="at">y =</span> variance_counts)) <span class="sc">+</span> </span>
<span id="cb14-3"><a href="differential-gene-expression-dge-analysis-overview.html#cb14-3" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x =</span> mean_counts, <span class="at">y =</span> mean_counts, <span class="at">color=</span><span class="st">&quot;red&quot;</span>)) <span class="sc">+</span></span>
<span id="cb14-4"><a href="differential-gene-expression-dge-analysis-overview.html#cb14-4" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span> <span class="fu">scale_x_log10</span>()</span></code></pre></div>
<p><img src="2024_RNA-seq-analysis_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<p>By plotting the <em>mean versus the variance</em> of our data we can easily see that the <strong>variance &gt; mean</strong> and therefore it does not fit the Poisson distribution. Genes having higher mean counts have even higher variance. Also for gene having low mean counts, there is a scatter of points and we can see that there is variability even in the variance. To account for this extra variance we need a new model.</p>
<p>The model that fits best, given this type of variability between replicates, is the <strong>Negative Binomial (NB) model</strong>. Essentially, <strong>the NB model</strong> is a good approximation for data where the <strong>variance &gt; mean</strong>, as is the case with RNA-seq count data.</p>
</div>
<div id="improving-mean-estimates-i.e.-reducing-variance-with-biological-replicates" class="section level3 hasAnchor" number="1.3.3">
<h3><span class="header-section-number">1.3.3</span> Improving mean estimates (i.e. reducing variance) with biological replicates<a href="differential-gene-expression-dge-analysis-overview.html#improving-mean-estimates-i.e.-reducing-variance-with-biological-replicates" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The variance or scatter tends to reduce as we increase the number of biological replicates (<em>the distribution will approach the Poisson distribution with increasing numbers of replicates</em>). <strong>The value of additional replicates is that as you add more data (replicates), you get increasingly precise estimates of group means, and ultimately greater confidence in the ability to distinguish differences between sample classes (i.e. more DE genes).</strong></p>
<p>The figure below shows how the number of differentially expressed (DE) genes identified is influenced by sequencing depth and number of replicates [<a href="https://academic.oup.com/bioinformatics/article/30/3/301/228651/RNA-seq-differential-expression-studies-more">1</a>]. Notice that increasing the number of replicates identifies more DE genes compared to increasing sequencing depth. This is because most of the biological variability occurs between samples. As a result, having more replicates typically yields better results than increasing sequencing depth, as it captures natural biological variability more effectively and leads to more accurate estimates of within-group variation. While increasing sequencing depth reduces technical noise, it has less impact on detecting DE genes since it doesn’t account for biological variability. Generally, a minimum sequencing depth of 20-30 million reads per sample is recommended, though successful RNA-seq experiments have been conducted with as few as 10 million reads when enough replicates are included.</p>
<p><img src="img/de_replicates_img.png" width="500" alt="Alt text" /></p>
<blockquote>
<p><strong>NOTE</strong></p>
<ul>
<li><strong>Biological replicates</strong> represent multiple samples (i.e. RNA from different mice) representing the same sample class</li>
<li><strong>Technical replicates</strong> represent the same sample (i.e. RNA from the same mouse) but with technical steps replicated</li>
<li>Usually biological variance is much greater than technical variance, so we do not need to account for technical variance to identify biological differences in expression</li>
<li><strong>Don’t spend money on technical replicates - biological replicates are much more useful</strong></li>
</ul>
</blockquote>
</div>
<div id="differential-expression-analysis-workflow" class="section level3 hasAnchor" number="1.3.4">
<h3><span class="header-section-number">1.3.4</span> Differential expression analysis workflow<a href="differential-gene-expression-dge-analysis-overview.html#differential-expression-analysis-workflow" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To model counts appropriately when performing a differential expression analysis, there are a number of software packages that have been developed for differential expression analysis of RNA-seq data. Even as new methods are continuously being developed a few tools are generally recommended as best practice, e.g. <strong><a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq2</a></strong> and <strong><a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html">EdgeR</a></strong>. Both these tools use the negative binomial model, employ similar methods, and typically, yield similar results. They are pretty stringent, and have a good balance between sensitivity and specificity (reducing both false positives and false negatives).</p>
<p>Here is a comparison of the three most highy used software packages for differential expression analysis. No single method is optimal under all circumstances, for example, limma works best when sample number is high, and edgeR and DESeq2 perform well for small sample sizes. It is also difficult to compare analysis methods due to different procedures in calculating pvalues.</p>
<p><img src="img/venn.png" width="300" alt="Alt text" /></p>
<p><strong>We will be using <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8">DESeq2</a> for the DE analysis, and the analysis steps with DESeq2 are shown in the flowchart below</strong>. DESeq2 first normalizes the count data to account for differences in library sizes and RNA composition between samples. Then, we will use the normalized counts to make some plots for QC at the gene and sample level. The final step is to use the appropriate functions from the DESeq2 package to perform the differential expression analysis.</p>
<p><img src="img/deseq_workflow_full_2018.png" width="500" alt="Alt text" /></p>
<p>We will go in-depth into each of these steps in the following lessons, but additional details and helpful suggestions regarding DESeq2 can be found in the <a href="http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html">DESeq2 vignette</a>. As you go through this workflow and questions arise, you can reference the vignette from within RStudio:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="differential-gene-expression-dge-analysis-overview.html#cb15-1" tabindex="-1"></a><span class="fu">vignette</span>(<span class="st">&quot;DESeq2&quot;</span>)</span></code></pre></div>
<p>This is very convenient, as it provides a wealth of information at your fingertips! Be sure to use this as you need during the workshop.</p>
<hr />

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="count-normalization.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["2024_RNA-seq-analysis.pdf", "2024_RNA-seq-analysis.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
